From 93820bd7ad4ea7db50221d411357c7212b6eebac Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Wed, 7 Aug 2013 19:39:41 -0400
Subject: [PATCH] Add API for vehicle exit event reason

---
 .../org/bukkit/event/vehicle/VehicleExitEvent.java     | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/main/java/org/bukkit/event/vehicle/VehicleExitEvent.java b/src/main/java/org/bukkit/event/vehicle/VehicleExitEvent.java
index a976c32de..ae50ecfbf 100644
--- a/src/main/java/org/bukkit/event/vehicle/VehicleExitEvent.java
+++ b/src/main/java/org/bukkit/event/vehicle/VehicleExitEvent.java
@@ -10,6 +10,16 @@ import org.jetbrains.annotations.NotNull;
  * Raised when a living entity exits a vehicle.
  */
 public class VehicleExitEvent extends VehicleEvent implements Cancellable {
+    // PaperDragon start
+    public enum DismountReason {
+        PLAYER, WATER, DEAD, DEAD_VEHICLE, TRANSFER, UNKNOWN, DISCONNECT;
+    }
+    public DismountReason reason = DismountReason.UNKNOWN;
+
+    public DismountReason getReason() {
+        return reason;
+    }
+    // PaperDragon end
     private static final HandlerList handlers = new HandlerList();
     private boolean cancelled;
     private final LivingEntity exited;
-- 
2.20.1.windows.1

