From 4b78710efbf1f092580ee19089dca27c6573c3fd Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Mon, 27 Jan 2020 00:56:50 -0500
Subject: [PATCH] Implement teleport() for offline players

---
 .../craftbukkit/CraftOfflinePlayer.java       | 28 +++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java b/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
index bd63f659fd..c4251e8e4d 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
@@ -568,5 +568,33 @@ public class CraftOfflinePlayer implements OfflinePlayer, ConfigurationSerializa
             return false;
         }
     }
+
+    @Override
+    public boolean teleport(Location loc) {
+        if (isOnline()) {
+            getPlayer().teleport(loc);
+            return true;
+        }
+        NBTTagCompound data = getData();
+
+        net.minecraft.nbt.NBTTagList rot = new net.minecraft.nbt.NBTTagList();
+        rot.add(0, net.minecraft.nbt.NBTTagFloat.a(loc.getYaw()));
+        rot.add(1, net.minecraft.nbt.NBTTagFloat.a(loc.getPitch()));
+        data.set("Rotation", rot);
+
+        net.minecraft.nbt.NBTTagList pos = new net.minecraft.nbt.NBTTagList();
+        pos.add(0, net.minecraft.nbt.NBTTagDouble.a(loc.getX()));
+        pos.add(1, net.minecraft.nbt.NBTTagDouble.a(loc.getY()));
+        pos.add(2, net.minecraft.nbt.NBTTagDouble.a(loc.getZ()));
+        data.set("Pos", pos);
+
+        UUID worldUUID = loc.getWorld().getUID();
+        data.set("WorldUUIDMost", net.minecraft.nbt.NBTTagLong.a(worldUUID.getMostSignificantBits()));
+        data.set("WorldUUIDLeast", net.minecraft.nbt.NBTTagLong.a(worldUUID.getLeastSignificantBits()));
+
+        data.setString("Dragon.DisconnectWorld", loc.getWorld().getName());
+
+        return saveData(data);
+    }
     // PaperDragon end
 }
-- 
2.20.1.windows.1

