From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Fri, 10 Jun 2016 11:22:21 -0400
Subject: [PATCH] PaperDragon Config Files


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index da5cab4246bd253fcc4e4d9574bdae1867ebb5ab..e16dfdade68bcb61676ea60f68ec60ca1505b524 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1948,6 +1948,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Paper end
 
+        // PaperDragon start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getDragonConfig() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // PaperDragon end
+
         /**
          * Sends the component to the player
          *
