From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Fri, 10 Jun 2016 11:22:21 -0400
Subject: [PATCH] PaperDragon Config Files


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 8621dba7f10ab822b5b99ce0d05da58823605cb6..334ea52133f20bce5cea531e350b24aa4be9a640 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1968,6 +1968,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Paper end
 
+        // PaperDragon start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getDragonConfig() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // PaperDragon end
+
         /**
          * Sends the component to the player
          *
