From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Fri, 10 Jun 2016 11:22:21 -0400
Subject: [PATCH] PaperDragon Config Files


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index c34cfba8f9ed7e9dbd0b6b8ffef9fba46abff046..6c6e695742a0f37d7ee6d2a5b7b92be0216a84cb 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1715,6 +1715,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Paper end
 
+        // PaperDragon start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getDragonConfig() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // PaperDragon end
+
         /**
          * Sends the component to the player
          *
