From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Fri, 10 Jun 2016 11:22:21 -0400
Subject: [PATCH] PaperDragon Config Files


diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index e756edf56995f4552387c2e1082307eb3dd48bb3..e07eae3367e60045d4c3434aaa921b65f09bc164 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1922,6 +1922,13 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
         }
         // Paper end
 
+        // PaperDragon start
+        @NotNull
+        public org.bukkit.configuration.file.YamlConfiguration getDragonConfig() {
+            throw new UnsupportedOperationException("Not supported yet.");
+        }
+        // PaperDragon end
+
         /**
          * Sends the component to the player
          *
