From 2c4a686abe2ddd6aed7d2b1b13d83042582140de Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Wed, 7 Aug 2013 19:39:41 -0400
Subject: [PATCH] Add API for vehicle exit event reason

---
 .../org/bukkit/event/vehicle/VehicleExitEvent.java    | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/main/java/org/bukkit/event/vehicle/VehicleExitEvent.java b/src/main/java/org/bukkit/event/vehicle/VehicleExitEvent.java
index a976c32d..6f0bf51b 100644
--- a/src/main/java/org/bukkit/event/vehicle/VehicleExitEvent.java
+++ b/src/main/java/org/bukkit/event/vehicle/VehicleExitEvent.java
@@ -10,6 +10,17 @@ import org.jetbrains.annotations.NotNull;
  * Raised when a living entity exits a vehicle.
  */
 public class VehicleExitEvent extends VehicleEvent implements Cancellable {
+    // PaperDragon start
+    public enum DismountReason {
+        PLAYER, WATER, DEAD, TRANSFER, UNKNOWN, DISCONNECT;
+    }
+    public DismountReason reason = DismountReason.UNKNOWN;
+
+    @NotNull
+    public DismountReason getReason() {
+        return reason;
+    }
+    // PaperDragon end
     private static final HandlerList handlers = new HandlerList();
     private boolean cancelled;
     private final LivingEntity exited;
-- 
2.22.0

