From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Wed, 7 Aug 2013 19:39:41 -0400
Subject: [PATCH] Add API for vehicle exit event reason


diff --git a/src/main/java/org/bukkit/event/vehicle/VehicleExitEvent.java b/src/main/java/org/bukkit/event/vehicle/VehicleExitEvent.java
index 39f6afd2f9cbcff6a74a91a21dcc3e29d2497dd8..ca07936552efe574ce24d7e1924850c4f73e2c8f 100644
--- a/src/main/java/org/bukkit/event/vehicle/VehicleExitEvent.java
+++ b/src/main/java/org/bukkit/event/vehicle/VehicleExitEvent.java
@@ -10,6 +10,17 @@ import org.jetbrains.annotations.NotNull;
  * Raised when a living entity exits a vehicle.
  */
 public class VehicleExitEvent extends VehicleEvent implements Cancellable {
+    // PaperDragon start
+    public enum DismountReason {
+        PLAYER, WATER, DEAD, DEAD_VEHICLE, TELEPORT, SPECTATE, LEASH, TRANSFER, UNKNOWN, DISCONNECT, FORCE
+    }
+    public DismountReason reason = DismountReason.UNKNOWN;
+
+    @NotNull
+    public DismountReason getReason() {
+        return reason;
+    }
+    // PaperDragon end
     private static final HandlerList handlers = new HandlerList();
     private boolean cancelled;
     private final LivingEntity exited;
