From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Wed, 3 Jun 2015 21:39:34 -0400
Subject: [PATCH] Ignore completion starting with _


diff --git a/src/main/java/org/bukkit/command/SimpleCommandMap.java b/src/main/java/org/bukkit/command/SimpleCommandMap.java
index 460fda05a62b12db2edcfb7ea8b2a5dd8e4b110d..d35c1ff09f9b8f15efb6cbca14d7633604f21e78 100644
--- a/src/main/java/org/bukkit/command/SimpleCommandMap.java
+++ b/src/main/java/org/bukkit/command/SimpleCommandMap.java
@@ -218,7 +218,7 @@ public class SimpleCommandMap implements CommandMap {
 
                 String name = commandEntry.getKey(); // Use the alias, not command name
 
-                if (StringUtil.startsWithIgnoreCase(name, cmdLine)) {
+                if (name.charAt(0) != '_' && StringUtil.startsWithIgnoreCase(name, cmdLine)) { // PaperDragon
                     completions.add(prefix + name);
                 }
             }
