From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Thu, 23 Jun 2016 22:22:09 -0400
Subject: [PATCH] User Object API

Allow Plugins to Store User Objects on the Player

diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 131daee2b29f7016463a00ce7927dff7b0a1b1b4..91961b831028bb03e29038c15b87f1a41e4a1dda 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -2652,4 +2652,12 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     @Override
     Spigot spigot();
     // Spigot end
+    
+    // PaperDragon start - User Object API
+    public void setUserObject(@NotNull org.bukkit.NamespacedKey key, @Nullable Object object);
+
+    @Nullable
+    public <T> T getUserObject(@NotNull org.bukkit.NamespacedKey key);
+    // PaperDragon end
+
 }
