From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Thu, 23 Jun 2016 22:22:09 -0400
Subject: [PATCH] User Object API

Allow Plugins to Store User Objects on the Player

diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index 1d98abff1ad0116f7a2599f078aa730cb84843c1..0e3433d939b9ec7bd38ffcc52e31cc905c569c6a 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -2832,4 +2832,12 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     @Override
     Spigot spigot();
     // Spigot end
+    
+    // PaperDragon start - User Object API
+    public void setUserObject(@NotNull org.bukkit.NamespacedKey key, @Nullable Object object);
+
+    @Nullable
+    public <T> T getUserObject(@NotNull org.bukkit.NamespacedKey key);
+    // PaperDragon end
+
 }
