From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Thu, 23 Jun 2016 22:22:09 -0400
Subject: [PATCH] User Object API

Allow Plugins to Store User Objects on the Player

diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index f954801f6a5d465b8545e75d7ff5af0352d6ec0d..5294c8a93ae291dce77a5a9d1646cd28e648d40b 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -2672,4 +2672,12 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     @Override
     Spigot spigot();
     // Spigot end
+    
+    // PaperDragon start - User Object API
+    public void setUserObject(@NotNull org.bukkit.NamespacedKey key, @Nullable Object object);
+
+    @Nullable
+    public <T> T getUserObject(@NotNull org.bukkit.NamespacedKey key);
+    // PaperDragon end
+
 }
