From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Thu, 23 Jun 2016 22:22:09 -0400
Subject: [PATCH] User Object API

Allow Plugins to Store User Objects on the Player

diff --git a/src/main/java/org/bukkit/entity/Player.java b/src/main/java/org/bukkit/entity/Player.java
index b607c229cfb1e95b17b6a0073380089ef5e1b675..1511d13f956ed8c97a07cd7268dce16afb71e723 100644
--- a/src/main/java/org/bukkit/entity/Player.java
+++ b/src/main/java/org/bukkit/entity/Player.java
@@ -2796,4 +2796,12 @@ public interface Player extends HumanEntity, Conversable, OfflinePlayer, PluginM
     @Override
     Spigot spigot();
     // Spigot end
+    
+    // PaperDragon start - User Object API
+    public void setUserObject(@NotNull org.bukkit.NamespacedKey key, @Nullable Object object);
+
+    @Nullable
+    public <T> T getUserObject(@NotNull org.bukkit.NamespacedKey key);
+    // PaperDragon end
+
 }
