From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Sat, 1 Apr 2023 11:29:02 -0500
Subject: [PATCH] Add Ranged Attack API


diff --git a/src/main/java/com/destroystokyo/paper/entity/RangedEntity.java b/src/main/java/com/destroystokyo/paper/entity/RangedEntity.java
index a087a04c525ea14b197c65063b43ebf097ddc5c2..8270fe24a08805f47cbbd8f52854a989a5507a16 100644
--- a/src/main/java/com/destroystokyo/paper/entity/RangedEntity.java
+++ b/src/main/java/com/destroystokyo/paper/entity/RangedEntity.java
@@ -3,6 +3,7 @@ package com.destroystokyo.paper.entity;
 import org.bukkit.entity.LivingEntity;
 import org.bukkit.entity.Mob;
 import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable; // PaperDragon
 
 public interface RangedEntity extends Mob {
     /**
@@ -32,4 +33,10 @@ public interface RangedEntity extends Mob {
     default boolean isChargingAttack() {
         return isHandRaised();
     }
+
+    // PaperDragon start
+    void setRangedAttackDistance(float range);
+
+    void setRangedAttackSpeed(@Nullable Integer min, @Nullable Integer max);
+    // PaperDragon end
 }
diff --git a/src/main/java/com/domnian/paperdragon/api/DAPI_Entity.java b/src/main/java/com/domnian/paperdragon/api/DAPI_Entity.java
index 6b0226d4afd02a77cb8380d26d21392bcbf0c809..fae7a41286a42e3791127be5b0acbb06666fd1bb 100644
--- a/src/main/java/com/domnian/paperdragon/api/DAPI_Entity.java
+++ b/src/main/java/com/domnian/paperdragon/api/DAPI_Entity.java
@@ -143,10 +143,14 @@ public interface DAPI_Entity {
     void setTargetRange(@NotNull Mob mob, @Nullable Double range);
 
     @Deprecated
-    void setRangedAttackDistance(@NotNull RangedEntity ranged, float range);
+    default void setRangedAttackDistance(@NotNull RangedEntity ranged, float range) {
+        ranged.setRangedAttackDistance(range);
+    }
 
     @Deprecated
-    void setRangedAttackSpeed(@NotNull RangedEntity ranged, @Nullable Integer min, @Nullable Integer max);
+    default void setRangedAttackSpeed(@NotNull RangedEntity ranged, @Nullable Integer min, @Nullable Integer max) {
+        ranged.setRangedAttackSpeed(min, max);
+    }
 
     @Deprecated
     void setAlwaysAngry(@NotNull Wolf wolf, boolean alwaysAngry);
