From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Sat, 1 Apr 2023 11:32:44 -0500
Subject: [PATCH] Add Wolf setAlwaysAngry API


diff --git a/src/main/java/org/bukkit/entity/Wolf.java b/src/main/java/org/bukkit/entity/Wolf.java
index 4b84c04675775e2a606630b00de8afe51665cebc..aa7aa0f770176e096d1a848ea8fc5a5611c38c42 100644
--- a/src/main/java/org/bukkit/entity/Wolf.java
+++ b/src/main/java/org/bukkit/entity/Wolf.java
@@ -74,6 +74,13 @@ public interface Wolf extends Tameable, Sittable, io.papermc.paper.entity.Collar
      */
     public void setInterested(boolean interested);
 
+    /**
+     * Set wolf to be always angry
+     *
+     * @param alwaysAngry Whether the wolf is always angry
+     */
+    void setAlwaysAngry(boolean alwaysAngry); // PaperDragon
+
     /**
      * Get the variant of this wolf.
      *
