From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Sat, 1 Apr 2023 11:48:20 -0500
Subject: [PATCH] Add Ghast API


diff --git a/src/main/java/org/bukkit/entity/Ghast.java b/src/main/java/org/bukkit/entity/Ghast.java
index 9c618a27d590f186f29c5d9094fc565efd40ca49..4857a9980ffa7dfeecebf067d0a18b32778f8751 100644
--- a/src/main/java/org/bukkit/entity/Ghast.java
+++ b/src/main/java/org/bukkit/entity/Ghast.java
@@ -35,4 +35,10 @@ public interface Ghast extends Flying, Enemy {
      */
     void setExplosionPower(int explosionPower);
     // Paper end
+
+    // PaperDragon start
+    int getFireballCooldown();
+
+    void setFireballCooldown(int cooldown);
+    // PaperDragon end
 }
