From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 3 Jul 2016 18:48:29 -0400
Subject: [PATCH] Expose Spigot's BungeeCord Setting to API


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index b5158151960db5c474b3aaa24bfa428e2a158fd9..449f64c85095d51debdf5ffaf079c12fd43bfe97 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -1982,4 +1982,13 @@ public final class Bukkit {
     public static Server.Spigot spigot() {
         return server.spigot();
     }
+
+    // PaperDragon start - Expose BungeeCord Setting
+    /**
+     * Get whether this server is in Bungeecord Mode or not.
+     */
+    public static boolean isBungeeEnabled() {
+        return server.isBungeeEnabled();
+    }
+    // PaperDragon end
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index d10e189892c5f40c576ed20bb0ea40bc1aff4490..4fddc0bc83c76509321fdefa30c84c75ac10f2a2 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1723,4 +1723,11 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
     @NotNull
     io.papermc.paper.datapack.DatapackManager getDatapackManager();
     // Paper end
+
+    // PaperDragon start - Expose BungeeCord Setting
+    /**
+     * Get whether this server is in Bungeecord Mode or not.
+     */
+    boolean isBungeeEnabled();
+    // PaperDragon end
 }
