From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 3 Jul 2016 18:48:29 -0400
Subject: [PATCH] Expose Spigot's BungeeCord Setting to API


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index 66580ade2cd8b8795b2d79f7958460162c3a17db..2d68b88ad75fbd8e38edcc407b9f7c8c065bbcaa 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2006,4 +2006,13 @@ public final class Bukkit {
     public static Server.Spigot spigot() {
         return server.spigot();
     }
+
+    // PaperDragon start - Expose BungeeCord Setting
+    /**
+     * Get whether this server is in Bungeecord Mode or not.
+     */
+    public static boolean isBungeeEnabled() {
+        return server.isBungeeEnabled();
+    }
+    // PaperDragon end
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 5d58b9790abda0a2b84701a38a4d42dbf746882c..cb689753f175ecad3ecdbde3cf669aad344584b1 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1743,4 +1743,11 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
     @NotNull
     io.papermc.paper.datapack.DatapackManager getDatapackManager();
     // Paper end
+
+    // PaperDragon start - Expose BungeeCord Setting
+    /**
+     * Get whether this server is in Bungeecord Mode or not.
+     */
+    boolean isBungeeEnabled();
+    // PaperDragon end
 }
