From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 3 Jul 2016 18:48:29 -0400
Subject: [PATCH] Expose Spigot's BungeeCord Setting to API


diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
index d36e4bc3c3713407704b865574cba28662f17315..eb91e469da523f0dd822498df500031109607cc9 100644
--- a/src/main/java/org/bukkit/Bukkit.java
+++ b/src/main/java/org/bukkit/Bukkit.java
@@ -2063,4 +2063,13 @@ public final class Bukkit {
     public static Server.Spigot spigot() {
         return server.spigot();
     }
+
+    // PaperDragon start - Expose BungeeCord Setting
+    /**
+     * Get whether this server is in Bungeecord Mode or not.
+     */
+    public static boolean isBungeeEnabled() {
+        return server.isBungeeEnabled();
+    }
+    // PaperDragon end
 }
diff --git a/src/main/java/org/bukkit/Server.java b/src/main/java/org/bukkit/Server.java
index 13640fb9236367104695fbf2b6134bd387c5a667..5ef6deecb52c3767327d602210a61fe5d117ba3a 100644
--- a/src/main/java/org/bukkit/Server.java
+++ b/src/main/java/org/bukkit/Server.java
@@ -1796,4 +1796,11 @@ public interface Server extends PluginMessageRecipient, net.kyori.adventure.audi
     @NotNull
     io.papermc.paper.datapack.DatapackManager getDatapackManager();
     // Paper end
+
+    // PaperDragon start - Expose BungeeCord Setting
+    /**
+     * Get whether this server is in Bungeecord Mode or not.
+     */
+    boolean isBungeeEnabled();
+    // PaperDragon end
 }
