From 4386bdc1aaa64e2617de298da433c2d3829cff84 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Thu, 3 May 2018 21:32:29 -0400
Subject: [PATCH] Enhanced Player Death Event

---
 .../bukkit/event/entity/PlayerDeathEvent.java | 21 +++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java b/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java
index f88dc3a0..0054a7ab 100644
--- a/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java
+++ b/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java
@@ -17,6 +17,7 @@ public class PlayerDeathEvent extends EntityDeathEvent {
     private boolean keepInventory = false;
     private List<ItemStack> itemsToKeep = new java.util.ArrayList(); // PaperDragon
     public List<ItemStack> getItemsToKeep() {return itemsToKeep;} // PaperDragon
+    private String sourceKey = "death.attack.generic"; // PaperDragon - Enhanced Player Death Event
 
     public PlayerDeathEvent(final Player player, final List<ItemStack> drops, final int droppedExp, final String deathMessage) {
         this(player, drops, droppedExp, 0, deathMessage);
@@ -160,4 +161,24 @@ public class PlayerDeathEvent extends EntityDeathEvent {
     public boolean getKeepInventory() {
         return keepInventory;
     }
+
+    // PaperDragon start - Enhanced Player Death Event
+    /**
+     * Sets the translation key for this type of death.
+     *
+     * @param key the translation key for this type of death.
+     */
+    public void setSourceKey(String key) {
+        this.sourceKey = key;
+    }
+
+    /**
+     * Gets the translation key for this type of death.
+     *
+     * @return the translation key for this type of death.
+     */
+    public String getSourceKey() {
+        return sourceKey;
+    }
+    // PaperDragon end
 }
-- 
2.17.0

