From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Thu, 3 May 2018 21:32:29 -0400
Subject: [PATCH] Enhanced Player Death Event


diff --git a/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java b/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java
index c7138d79a572a525c70b51b7c79d2c5dfb18f25c..ffae02576947b789186d6dde2345a4146fe21710 100644
--- a/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java
+++ b/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java
@@ -19,6 +19,7 @@ public class PlayerDeathEvent extends EntityDeathEvent {
     private int newTotalExp = 0;
     private boolean keepLevel = false;
     private boolean keepInventory = false;
+    private String sourceKey = "death.attack.generic"; // PaperDragon - Enhanced Player Death Event
     // Paper start
     private boolean doExpDrop;
 
@@ -291,4 +292,26 @@ public class PlayerDeathEvent extends EntityDeathEvent {
     public boolean getKeepInventory() {
         return keepInventory;
     }
+
+    // PaperDragon start - Enhanced Player Death Event
+    /**
+     * Sets the translation key for this type of death.
+     *
+     * @param key the translation key for this type of death.
+     */
+    public void setSourceKey(@NotNull String key) {
+        this.sourceKey = key;
+    }
+
+    /**
+     * Gets the translation key for this type of death.
+     *
+     * @return the translation key for this type of death.
+     */
+    @NotNull
+    public String getSourceKey() {
+        return sourceKey;
+    }
+    // PaperDragon end
+
 }
