From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Wed, 10 Jul 2019 04:47:04 -0400
Subject: [PATCH] Add method to get player skin hash


diff --git a/src/main/java/com/domnian/paperdragon/api/DAPI_Misc.java b/src/main/java/com/domnian/paperdragon/api/DAPI_Misc.java
index 4600c91e958e349622ae740385bae39f5cca02c4..e1a1066f0d4aa59cdcc32feb2ddc76113a31c707 100644
--- a/src/main/java/com/domnian/paperdragon/api/DAPI_Misc.java
+++ b/src/main/java/com/domnian/paperdragon/api/DAPI_Misc.java
@@ -25,6 +25,7 @@ package com.domnian.paperdragon.api;
 
 import org.bukkit.Location;
 import org.bukkit.StructureType;
+import org.bukkit.entity.Player;
 import org.bukkit.inventory.ItemStack;
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
@@ -51,4 +52,11 @@ public interface DAPI_Misc {
     @Nullable
     List<ItemStack> getShulkerInventory(@NotNull ItemStack shulker);
 
+    /**
+     * Fetches a Player's skin hash
+     * @param player
+     * @return player's skin hash
+     */
+    String getSkinHash(@NotNull Player player);
+
 }
