From 191b996d5be51cc39928e9fab0dfe89403cb3d40 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Thu, 3 May 2018 21:32:29 -0400
Subject: [PATCH] Enhanced Player Death Event

---
 .../bukkit/event/entity/PlayerDeathEvent.java | 22 +++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java b/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java
index f5c3c5e2..784f33fd 100644
--- a/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java
+++ b/src/main/java/org/bukkit/event/entity/PlayerDeathEvent.java
@@ -16,6 +16,7 @@ public class PlayerDeathEvent extends EntityDeathEvent {
     private int newTotalExp = 0;
     private boolean keepLevel = false;
     private boolean keepInventory = false;
+    private String sourceKey = "death.attack.generic"; // PaperDragon - Enhanced Player Death Event
 
     // Paper start
     private List<ItemStack> itemsToKeep = new java.util.ArrayList<>();
@@ -195,4 +196,25 @@ public class PlayerDeathEvent extends EntityDeathEvent {
     public boolean getKeepInventory() {
         return keepInventory;
     }
+
+    // PaperDragon start - Enhanced Player Death Event
+    /**
+     * Sets the translation key for this type of death.
+     *
+     * @param key the translation key for this type of death.
+     */
+    public void setSourceKey(@NotNull String key) {
+        this.sourceKey = key;
+    }
+
+    /**
+     * Gets the translation key for this type of death.
+     *
+     * @return the translation key for this type of death.
+     */
+    @NotNull
+    public String getSourceKey() {
+        return sourceKey;
+    }
+    // PaperDragon end
 }
-- 
2.17.1

