From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Mon, 27 Jan 2020 00:56:34 -0500
Subject: [PATCH] Add teleport() to OfflinePlayer


diff --git a/src/main/java/org/bukkit/OfflinePlayer.java b/src/main/java/org/bukkit/OfflinePlayer.java
index 08861bb33bb25d1a1755c38d98ce2ed26cad26f5..3e0081b8861b8fd635153c71ace072d47cf8b600 100644
--- a/src/main/java/org/bukkit/OfflinePlayer.java
+++ b/src/main/java/org/bukkit/OfflinePlayer.java
@@ -459,5 +459,6 @@ public interface OfflinePlayer extends ServerOperator, AnimalTamer, Configuratio
     // PaperDragon start
     @NotNull
     public Location getLocation();
+    public boolean teleport(@NotNull Location location);
     // PaperDragon end
 }
