From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Mon, 27 Jan 2020 00:56:34 -0500
Subject: [PATCH] Add teleport() to OfflinePlayer


diff --git a/src/main/java/org/bukkit/OfflinePlayer.java b/src/main/java/org/bukkit/OfflinePlayer.java
index 5827faf0b12e600e810f60b8851db3e2e66cb370..0d0ca91393b70b70940c25ca16ccc5be267cf7cc 100644
--- a/src/main/java/org/bukkit/OfflinePlayer.java
+++ b/src/main/java/org/bukkit/OfflinePlayer.java
@@ -459,5 +459,6 @@ public interface OfflinePlayer extends ServerOperator, AnimalTamer, Configuratio
     // PaperDragon start
     @NotNull
     public Location getLocation();
+    public boolean teleport(@NotNull Location location);
     // PaperDragon end
 }
