From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Mon, 27 Jan 2020 00:56:34 -0500
Subject: [PATCH] Add teleport() to OfflinePlayer


diff --git a/src/main/java/org/bukkit/OfflinePlayer.java b/src/main/java/org/bukkit/OfflinePlayer.java
index c623862c9c0bc9a896313f74beb08e96f1d206c5..6589636f159a1b1c5837fc8bbda5fa49f97e7f5e 100644
--- a/src/main/java/org/bukkit/OfflinePlayer.java
+++ b/src/main/java/org/bukkit/OfflinePlayer.java
@@ -451,5 +451,6 @@ public interface OfflinePlayer extends ServerOperator, AnimalTamer, Configuratio
     // PaperDragon start
     @NotNull
     public Location getLocation();
+    public boolean teleport(@NotNull Location location);
     // PaperDragon end
 }
