From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Mon, 27 Jan 2020 00:56:34 -0500
Subject: [PATCH] Add teleport() to OfflinePlayer


diff --git a/src/main/java/org/bukkit/OfflinePlayer.java b/src/main/java/org/bukkit/OfflinePlayer.java
index 4bb55781b59be4b83143695849c9c50b6b02d410..a2d5b2af4e93749febc26829240e3f392d573143 100644
--- a/src/main/java/org/bukkit/OfflinePlayer.java
+++ b/src/main/java/org/bukkit/OfflinePlayer.java
@@ -438,5 +438,6 @@ public interface OfflinePlayer extends ServerOperator, AnimalTamer, Configuratio
     // PaperDragon start
     @NotNull
     public Location getLocation();
+    public boolean teleport(@NotNull Location location);
     // PaperDragon end
 }
