From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Sun, 3 Jan 2016 22:02:53 -0500
Subject: [PATCH] Add UnknownCommandEvent


diff --git a/src/main/java/com/domnian/paperdragon/events/UnknownCommandEvent.java b/src/main/java/com/domnian/paperdragon/events/UnknownCommandEvent.java
new file mode 100644
index 0000000000000000000000000000000000000000..6e81766537fe825962c9490b36e19c7097cc24d7
--- /dev/null
+++ b/src/main/java/com/domnian/paperdragon/events/UnknownCommandEvent.java
@@ -0,0 +1,58 @@
+package com.domnian.paperdragon.events;
+
+import org.bukkit.command.CommandSender;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import org.jetbrains.annotations.NotNull;
+
+public class UnknownCommandEvent extends Event implements Cancellable {
+    private static final HandlerList handlers = new HandlerList();
+
+    private final CommandSender sender;
+    private final String sentCommandLabel;
+    private final String commandLine;
+
+    public UnknownCommandEvent(@NotNull CommandSender sender, @NotNull String sentCommandLabel, @NotNull String commandLine) {
+        this.sender = sender;
+        this.sentCommandLabel = sentCommandLabel;
+        this.commandLine = commandLine;
+    }
+
+    @NotNull
+    public CommandSender getSender() {
+        return sender;
+    }
+
+    @NotNull
+    public String getSentCommandLabel() {
+        return sentCommandLabel;
+    }
+
+    @NotNull
+    public String getCommandLine() {
+        return commandLine;
+    }
+
+    @NotNull
+    public HandlerList getHandlers() {
+        return handlers;
+    }
+
+    @NotNull
+    public static HandlerList getHandlerList() {
+        return handlers;
+    }
+
+    private boolean cancelled = false;
+
+    @Override
+    public boolean isCancelled() {
+        return cancelled;
+    }
+
+    @Override
+    public void setCancelled(boolean cancel) {
+        cancelled = cancel;
+    }
+}
diff --git a/src/main/java/org/bukkit/command/SimpleCommandMap.java b/src/main/java/org/bukkit/command/SimpleCommandMap.java
index a4962b329a921078aabee31295111b9639dc4639..9c694140f3ea808868e6c59cd4a3e707ae745527 100644
--- a/src/main/java/org/bukkit/command/SimpleCommandMap.java
+++ b/src/main/java/org/bukkit/command/SimpleCommandMap.java
@@ -144,7 +144,7 @@ public class SimpleCommandMap implements CommandMap {
         Command target = getCommand(sentCommandLabel);
 
         if (target == null) {
-            return false;
+            return !(new com.domnian.paperdragon.events.UnknownCommandEvent(sender, sentCommandLabel, commandLine).callEvent()); // PaperDragon
         }
 
         // Paper start - Plugins do weird things to workaround normal registration
