From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 27 Jun 2021 11:44:05 -0400
Subject: [PATCH] Add method to serialize an ItemStack's NBT tag


diff --git a/src/main/java/com/domnian/paperdragon/api/DAPI_Misc.java b/src/main/java/com/domnian/paperdragon/api/DAPI_Misc.java
index fe23d1ae53230dd326f5ca5a80b8a84716401785..e5a22552ec06e924c65c1bb6090ab9f5c0684590 100644
--- a/src/main/java/com/domnian/paperdragon/api/DAPI_Misc.java
+++ b/src/main/java/com/domnian/paperdragon/api/DAPI_Misc.java
@@ -58,4 +58,7 @@ public interface DAPI_Misc {
      */
     String getSkinHash(@NotNull Player player);
 
+    @NotNull
+    String getSerializedItemTag(@NotNull ItemStack stack);
+
 }
