From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sat, 20 Nov 2021 17:20:55 -0500
Subject: [PATCH] Add HandlerList#hasListeners() utility method


diff --git a/src/main/java/org/bukkit/event/HandlerList.java b/src/main/java/org/bukkit/event/HandlerList.java
index 2292bd460ce2be113beb4ba6b4eb19350060f01c..ea991b0c5a39f5fcccf80de37c26f6a5117efb9f 100644
--- a/src/main/java/org/bukkit/event/HandlerList.java
+++ b/src/main/java/org/bukkit/event/HandlerList.java
@@ -210,6 +210,17 @@ public class HandlerList {
         return handlers;
     }
 
+    // PaperDragon start
+    /**
+     * Check whether this handler list has any registered listeners
+     *
+     * @return true if there are any registered listeners
+     */
+    public boolean hasListeners() {
+        return getRegisteredListeners().length > 0;
+    }
+    // PaperDragon end
+
     /**
      * Get a specific plugin's registered listeners associated with this
      * handler list
