From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 2 Jan 2022 10:10:22 -0500
Subject: [PATCH] Add methods to get a raid's id and to cancel it


diff --git a/src/main/java/org/bukkit/Raid.java b/src/main/java/org/bukkit/Raid.java
index a8cccd707a0c353cef14e8c5aabc0498f686d934..858ed6d289f1b97d5e114794a70e32dc4254ceb7 100644
--- a/src/main/java/org/bukkit/Raid.java
+++ b/src/main/java/org/bukkit/Raid.java
@@ -110,6 +110,11 @@ public interface Raid extends org.bukkit.persistence.PersistentDataHolder { // P
     @NotNull
     List<Raider> getRaiders();
 
+    // PaperDragon start
+    int getId();
+    void cancel();
+    // PaperDragon end
+
     /**
      * Represents the status of a {@link Raid}.
      */
