From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 2 Jan 2022 10:10:22 -0500
Subject: [PATCH] Add methods to get a raid's id and to cancel it


diff --git a/src/main/java/org/bukkit/Raid.java b/src/main/java/org/bukkit/Raid.java
index 29a0067600bdecdcccdc8f65b21db13d093c032c..78fc60c6380c6f8c84acf08aa9513a38239244f3 100644
--- a/src/main/java/org/bukkit/Raid.java
+++ b/src/main/java/org/bukkit/Raid.java
@@ -109,6 +109,11 @@ public interface Raid {
     @NotNull
     List<Raider> getRaiders();
 
+    // PaperDragon start
+    int getId();
+    void cancel();
+    // PaperDragon end
+
     /**
      * Represents the status of a {@link Raid}.
      */
