From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 20 Jun 2021 15:18:46 -0400
Subject: [PATCH] Build Changes


diff --git a/build.gradle.kts b/build.gradle.kts
index 39beb0331282b297f4f3857560d24ce6a616210d..7764dcc7caebc9f12a6775ae1aec2c2da49d24ae 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -30,8 +30,8 @@ repositories {
 }
 
 dependencies {
-    implementation(project(":Paper-API"))
-    implementation(project(":Paper-MojangAPI"))
+    implementation(project(":PaperDragon-API")) // PaperDragon
+    implementation("io.papermc.paper:paper-mojangapi:1.17-R0.1-SNAPSHOT") // PaperDragon
     // Paper start
     implementation("org.jline:jline-terminal-jansi:3.12.1")
     implementation("net.minecrell:terminalconsoleappender:1.2.0")
@@ -66,6 +66,12 @@ dependencies {
 
     implementation("org.quiltmc:tiny-mappings-parser:0.3.0") // Paper - needed to read mappings for stacktrace deobfuscation
 
+    // PaperDragon - start
+    runtimeOnly("com.zaxxer:HikariCP:4.0.3")
+    runtimeOnly("net.minidev:json-smart:2.4.7")
+    runtimeOnly("org.json:json:20210307")
+    // PaperDragon - end
+
     testImplementation("io.github.classgraph:classgraph:4.8.47") // Paper - mob goal test
     testImplementation("junit:junit:4.13.1")
     testImplementation("org.hamcrest:hamcrest-library:1.3")
@@ -82,7 +88,7 @@ tasks.jar {
         attributes(
             "Main-Class" to "org.bukkit.craftbukkit.Main",
             "Implementation-Title" to "CraftBukkit",
-            "Implementation-Version" to "git-Paper-$implementationVersion",
+            "Implementation-Version" to "git-PaperDragon-$implementationVersion", // PaperDragon
             "Implementation-Vendor" to SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'").format(Date()), // Paper
             "Specification-Title" to "Bukkit",
             "Specification-Version" to project.version,
@@ -114,7 +120,7 @@ tasks.shadowJar {
 
     // Needed for Paperclip's install to maven local feature
     from(generatePom) {
-        into("META-INF/maven/io.papermc.paper/paper")
+        into("META-INF/maven/com.domnian.paperdragon/paperdragon")
         rename { "pom.xml" }
     }
 
@@ -178,7 +184,7 @@ tasks.test {
 fun TaskContainer.registerRunTask(
     name: String, block: JavaExec.() -> Unit
 ): TaskProvider<JavaExec> = register<JavaExec>(name) {
-    group = "paper"
+    group = "paperdragon"
     standardInput = System.`in`
     workingDir = rootProject.layout.projectDirectory.dir(
         providers.gradleProperty("runWorkDir").forUseAtConfigurationTime().orElse("run")
diff --git a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
index 774556a62eb240da42e84db4502e2ed43495be17..50cd5f9cacb4cf5f6c1a28fc7bf17321ef18a868 100644
--- a/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
+++ b/src/main/java/org/bukkit/craftbukkit/util/Versioning.java
@@ -11,7 +11,7 @@ public final class Versioning {
     public static String getBukkitVersion() {
         String result = "Unknown-Version";
 
-        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/io.papermc.paper/paper-api/pom.properties");
+        InputStream stream = Bukkit.class.getClassLoader().getResourceAsStream("META-INF/maven/com.domnian.paperdragon/paperdragon-api/pom.properties");
         Properties properties = new Properties();
 
         if (stream != null) {
