From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Wed, 7 Aug 2013 19:52:14 -0400
Subject: [PATCH] VehicleExitEvent Dismount Reason API


diff --git a/src/main/java/net/minecraft/server/commands/TeleportCommand.java b/src/main/java/net/minecraft/server/commands/TeleportCommand.java
index ced3b3ed227dacf1c4928ee2dfa4f390f13c8a8b..6c4affb6bf8efbf07168c841eea9ddd55da68daa 100644
--- a/src/main/java/net/minecraft/server/commands/TeleportCommand.java
+++ b/src/main/java/net/minecraft/server/commands/TeleportCommand.java
@@ -38,6 +38,7 @@ import net.minecraft.world.phys.Vec3;
 import org.bukkit.Location;
 import org.bukkit.craftbukkit.CraftWorld;
 import org.bukkit.event.entity.EntityTeleportEvent;
+import org.bukkit.event.vehicle.VehicleExitEvent; // PaperDragon
 // CraftBukkit end
 
 public class TeleportCommand {
@@ -158,6 +159,7 @@ public class TeleportCommand {
                 ChunkPos chunkcoordintpair = new ChunkPos(new BlockPos(x, y, z));
 
                 world.getChunkSource().addRegionTicket(TicketType.POST_TELEPORT, chunkcoordintpair, 1, target.getId());
+                target.dismountReason = VehicleExitEvent.DismountReason.TELEPORT; // PaperDragon
                 target.stopRiding();
                 if (((ServerPlayer) target).isSleeping()) {
                     ((ServerPlayer) target).stopSleepInBed(true, true);
diff --git a/src/main/java/net/minecraft/server/level/ServerLevel.java b/src/main/java/net/minecraft/server/level/ServerLevel.java
index c2e72761fd0ec5dd0ed4817eb376283bd5d715a7..9b64ae0ca95333e48b06128d8b52cfab5d159592 100644
--- a/src/main/java/net/minecraft/server/level/ServerLevel.java
+++ b/src/main/java/net/minecraft/server/level/ServerLevel.java
@@ -156,6 +156,7 @@ import org.bukkit.craftbukkit.event.CraftEventFactory;
 import org.bukkit.craftbukkit.util.WorldUUID;
 import org.bukkit.event.entity.CreatureSpawnEvent;
 import org.bukkit.event.server.MapInitializeEvent;
+import org.bukkit.event.vehicle.VehicleExitEvent; // PaperDragon
 import org.bukkit.event.weather.LightningStrikeEvent;
 import org.bukkit.event.world.TimeSkipEvent;
 // CraftBukkit end
@@ -661,6 +662,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
                                     return;
                                 }
 
+                                if (entity1.isRemoved()) entity.dismountReason =  VehicleExitEvent.DismountReason.DEAD_VEHICLE; // PaperDragon
                                 entity.stopRiding();
                             }
 
@@ -1186,6 +1188,7 @@ public class ServerLevel extends Level implements WorldGenLevel {
             } finally { timer.stopTiming(); }// Paper - EAR2 timings
             }
         } else {
+            if (passenger.isRemoved()) { passenger.dismountReason = VehicleExitEvent.DismountReason.DEAD; } // PaperDragon
             passenger.stopRiding();
         }
     }
diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
index b6f9c99d580d985f2b84efaa655d9ae40073e72b..91b0cdd153eb183c83d4d947a83b6dc6a0882266 100644
--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
+++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
@@ -163,6 +163,7 @@ import org.bukkit.event.player.PlayerLocaleChangeEvent;
 import org.bukkit.event.player.PlayerPortalEvent;
 import org.bukkit.event.player.PlayerTeleportEvent;
 import org.bukkit.event.player.PlayerTeleportEvent.TeleportCause;
+import org.bukkit.event.vehicle.VehicleExitEvent; // PaperDragon
 import org.bukkit.inventory.MainHand;
 // CraftBukkit end
 
@@ -1661,10 +1662,11 @@ public class ServerPlayer extends Player {
 
     public void disconnect() {
         this.disconnected = true;
-        this.ejectPassengers();
+        this.ejectPassengers(VehicleExitEvent.DismountReason.DISCONNECT); // PaperDragon
 
         // Paper start - Workaround an issue where the vehicle doesn't track the passenger disconnection dismount.
         if (this.isPassenger() && this.getVehicle() instanceof ServerPlayer) {
+            this.dismountReason = VehicleExitEvent.DismountReason.DISCONNECT; // PaperDragon
             this.stopRiding();
         }
         // Paper end
@@ -1830,6 +1832,7 @@ public class ServerPlayer extends Player {
             this.connection.send(new ClientboundGameEventPacket(ClientboundGameEventPacket.CHANGE_GAME_MODE, (float) gameMode.getId()));
             if (gameMode == GameType.SPECTATOR) {
                 this.removeEntitiesOnShoulder();
+                this.dismountReason = org.bukkit.event.vehicle.VehicleExitEvent.DismountReason.SPECTATE; // PaperDragon
                 this.stopRiding();
             } else {
                 this.setCamera(this);
@@ -2004,7 +2007,7 @@ public class ServerPlayer extends Player {
 
         if (entity != this) {
             // Make sure we're in the right place
-            this.ejectPassengers(); // teleport can fail if we have passengers...
+            this.ejectPassengers(VehicleExitEvent.DismountReason.SPECTATE); // teleport can fail if we have passengers... // PaperDragon
             this.getBukkitEntity().teleport(new Location(entity.getCommandSenderWorld().getWorld(), entity.getX(), entity.getY(), entity.getZ(), this.getYRot(), this.getXRot()), TeleportCause.SPECTATE); // Correctly handle cross-world entities from api calls by using CB teleport
 
             // Make sure we're tracking the entity before sending
@@ -2072,6 +2075,7 @@ public class ServerPlayer extends Player {
     public void teleportTo(ServerLevel worldserver, double d0, double d1, double d2, float f, float f1, org.bukkit.event.player.PlayerTeleportEvent.TeleportCause cause) {
         // CraftBukkit end
         this.setCamera(this);
+        this.dismountReason = VehicleExitEvent.DismountReason.TELEPORT; // PaperDragon
         this.stopRiding();
         /* CraftBukkit start - replace with bukkit handling for multi-world
         if (worldserver == this.level) {
diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
index 40d55cade76d0e8f39f91a322bc8a821877efa2c..78fb7e0a568f0746af803cbe81247f5e680ae685 100644
--- a/src/main/java/net/minecraft/server/players/PlayerList.java
+++ b/src/main/java/net/minecraft/server/players/PlayerList.java
@@ -118,6 +118,7 @@ import org.bukkit.event.player.PlayerJoinEvent;
 import org.bukkit.event.player.PlayerLoginEvent;
 import org.bukkit.event.player.PlayerQuitEvent;
 import org.bukkit.event.player.PlayerRespawnEvent;
+import org.bukkit.event.vehicle.VehicleExitEvent; // PaperDragon
 // CraftBukkit end
 
 public abstract class PlayerList {
@@ -633,6 +634,7 @@ public abstract class PlayerList {
 
             if (entity.hasExactlyOnePlayerPassenger()) {
                 PlayerList.LOGGER.debug("Removing player mount");
+                entityplayer.dismountReason = VehicleExitEvent.DismountReason.DISCONNECT; // PaperDragon
                 entityplayer.stopRiding();
                 entity.getPassengersAndSelf().forEach((entity1) -> {
                     // Paper start
diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 08a74f4ff92cf111547248ca370856ba3e1c90a1..f34bd787d28706e0059e6a99d65ff9e6a9f976bb 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -729,6 +729,7 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
         this.level.getProfiler().push("entityBaseTick");
         this.feetBlockState = null;
         if (this.isPassenger() && this.getVehicle().isRemoved()) {
+            this.dismountReason = VehicleExitEvent.DismountReason.DEAD_VEHICLE; // PaperDragon
             this.stopRiding();
         }
 
@@ -2451,6 +2452,7 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
                 return false;
             } else {
                 if (this.isPassenger()) {
+                    this.dismountReason = VehicleExitEvent.DismountReason.TRANSFER; // PaperDragon
                     this.stopRiding();
                 }
 
@@ -2476,7 +2478,13 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
     }
 
     public void ejectPassengers() {
+        // PaperDragon start - allow passing reason to ejectPassengers()
+        ejectPassengers(VehicleExitEvent.DismountReason.UNKNOWN);
+    }
+    public void ejectPassengers(VehicleExitEvent.DismountReason reason) {
+        // PaperDragon end
         for (int i = this.passengers.size() - 1; i >= 0; --i) {
+            this.passengers.get(i).dismountReason = reason; // PaperDragon
             ((Entity) this.passengers.get(i)).stopRiding();
         }
 
@@ -2558,6 +2566,8 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
     // Paper start
     protected boolean removePassenger(Entity entity) { return removePassenger(entity, false);}
     protected boolean removePassenger(Entity entity, boolean suppressCancellation) { // CraftBukkit
+        VehicleExitEvent.DismountReason reason = entity.dismountReason; // PaperDragon
+        entity.dismountReason = VehicleExitEvent.DismountReason.UNKNOWN; // PaperDragon
         // Paper end
         if (entity.getVehicle() == this) {
             throw new IllegalStateException("Use x.stopRiding(y), not y.removePassenger(x)");
@@ -2570,6 +2580,7 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
                         (Vehicle) this.getBukkitEntity(),
                         (LivingEntity) entity.getBukkitEntity(), !suppressCancellation // Paper
                 );
+                event.reason = reason; // PaperDragon
                 // Suppress during worldgen
                 if (this.valid) {
                     Bukkit.getPluginManager().callEvent(event);
@@ -3071,7 +3082,7 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
 
         return this.removalReason != null ? String.format(Locale.ROOT, "%s['%s'/%d, uuid='%s', l='%s', x=%.2f, y=%.2f, z=%.2f, cpos=%s, tl=%d, v=%b, removed=%s]", this.getClass().getSimpleName(), this.getName().getString(), this.id, this.uuid, s, this.getX(), this.getY(), this.getZ(), this.chunkPosition(), this.tickCount, this.valid, this.removalReason) : String.format(Locale.ROOT, "%s['%s'/%d, uuid='%s', l='%s', x=%.2f, y=%.2f, z=%.2f, cpos=%s, tl=%d, v=%b]", this.getClass().getSimpleName(), this.getName().getString(), this.id, this.uuid, s, this.getX(), this.getY(), this.getZ(), this.chunkPosition(), this.tickCount, this.valid);
     }
-
+    public VehicleExitEvent.DismountReason dismountReason = VehicleExitEvent.DismountReason.UNKNOWN; // PaperDragon
     public boolean isInvulnerableTo(DamageSource damageSource) {
         return this.isRemoved() || this.invulnerable && damageSource != DamageSource.OUT_OF_WORLD && !damageSource.isCreativePlayer() && !isDisabled; // PaperDragon - add !isDisabled
     }
@@ -4088,10 +4099,16 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
         }
 
         if (this.removalReason.shouldDestroy()) {
+            this.dismountReason = VehicleExitEvent.DismountReason.DEAD; // PaperDragon
             this.stopRiding();
         }
 
-        this.getPassengers().forEach(Entity::stopRiding);
+        // PaperDragon start
+        this.getPassengers().forEach(entity -> {
+            entity.dismountReason = VehicleExitEvent.DismountReason.DEAD_VEHICLE;
+            entity.stopRiding();
+        });
+        // PaperDragon end
         this.levelCallback.onRemove(reason);
     }
 
diff --git a/src/main/java/net/minecraft/world/entity/LivingEntity.java b/src/main/java/net/minecraft/world/entity/LivingEntity.java
index cbd893ecbfe1d9583423118d34d655a58097e7e7..bc28cc08d6fd62dc0687f60cda8651b78a0f78f9 100644
--- a/src/main/java/net/minecraft/world/entity/LivingEntity.java
+++ b/src/main/java/net/minecraft/world/entity/LivingEntity.java
@@ -431,6 +431,7 @@ public abstract class LivingEntity extends Entity {
                 }
 
                 if (!this.level.isClientSide && this.isPassenger() && this.getVehicle() != null && !this.getVehicle().rideableUnderWater()) {
+                    this.dismountReason = org.bukkit.event.vehicle.VehicleExitEvent.DismountReason.WATER; // PaperDragon
                     this.stopRiding();
                 }
             } else if (this.getAirSupply() < this.getMaxAirSupply()) {
@@ -4101,6 +4102,7 @@ public abstract class LivingEntity extends Entity {
 
     public void startSleeping(BlockPos pos) {
         if (this.isPassenger()) {
+            if (this instanceof net.minecraft.world.entity.player.Player) this.dismountReason = org.bukkit.event.vehicle.VehicleExitEvent.DismountReason.PLAYER; // PaperDragon
             this.stopRiding();
         }
 
diff --git a/src/main/java/net/minecraft/world/entity/Mob.java b/src/main/java/net/minecraft/world/entity/Mob.java
index 01c25c440ce147e5a204ceb68175bb36e0a124d1..e5845c04a5519bfddfba22ad31e2c69d599abee4 100644
--- a/src/main/java/net/minecraft/world/entity/Mob.java
+++ b/src/main/java/net/minecraft/world/entity/Mob.java
@@ -81,6 +81,7 @@ import org.bukkit.event.entity.EntityTargetEvent;
 import org.bukkit.event.entity.EntityTransformEvent;
 import org.bukkit.event.entity.EntityUnleashEvent;
 import org.bukkit.event.entity.EntityUnleashEvent.UnleashReason;
+import org.bukkit.event.vehicle.VehicleExitEvent; // PaperDragon
 // CraftBukkit end
 
 public abstract class Mob extends LivingEntity {
@@ -1453,6 +1454,7 @@ public abstract class Mob extends LivingEntity {
         }
 
         if (this.isPassenger()) {
+            this.dismountReason = VehicleExitEvent.DismountReason.LEASH; // PaperDragon
             this.stopRiding();
         }
 
diff --git a/src/main/java/net/minecraft/world/entity/player/Player.java b/src/main/java/net/minecraft/world/entity/player/Player.java
index 2c00a766130a7f682fc6c4c74321e10637ca7932..6ff1c8d873a6b8b3c71fa4f696cdc336ca35b3cf 100644
--- a/src/main/java/net/minecraft/world/entity/player/Player.java
+++ b/src/main/java/net/minecraft/world/entity/player/Player.java
@@ -509,6 +509,7 @@ public abstract class Player extends LivingEntity {
     @Override
     public void rideTick() {
         if (!this.level.isClientSide && this.wantsToStopRiding() && this.isPassenger()) {
+            this.dismountReason = org.bukkit.event.vehicle.VehicleExitEvent.DismountReason.PLAYER; // PaperDragon
             this.stopRiding();
             this.setShiftKeyDown(false);
         } else {
diff --git a/src/main/java/net/minecraft/world/entity/vehicle/AbstractMinecart.java b/src/main/java/net/minecraft/world/entity/vehicle/AbstractMinecart.java
index ad49dcc3473fbad306d21cbac4600574e80220a7..bef5981ffe165dd7fb7d5364b4b65c55f9a87243 100644
--- a/src/main/java/net/minecraft/world/entity/vehicle/AbstractMinecart.java
+++ b/src/main/java/net/minecraft/world/entity/vehicle/AbstractMinecart.java
@@ -54,6 +54,7 @@ import org.bukkit.entity.Vehicle;
 import org.bukkit.event.vehicle.VehicleDamageEvent;
 import org.bukkit.event.vehicle.VehicleDestroyEvent;
 import org.bukkit.event.vehicle.VehicleEntityCollisionEvent;
+import org.bukkit.event.vehicle.VehicleExitEvent; // PaperDragon
 import org.bukkit.util.Vector;
 // CraftBukkit end
 
@@ -266,7 +267,7 @@ public abstract class AbstractMinecart extends Entity {
                         return true;
                     }
                     // CraftBukkit end
-                    this.ejectPassengers();
+                    this.ejectPassengers(VehicleExitEvent.DismountReason.DEAD_VEHICLE); // PaperDragon
                     if (flag && !this.hasCustomName()) {
                         this.discard();
                     } else {
diff --git a/src/main/java/net/minecraft/world/item/ChorusFruitItem.java b/src/main/java/net/minecraft/world/item/ChorusFruitItem.java
index 062881d0b3d57401dc4f6d876b802699df586574..15bdc9f6be51fb16b5b9d2dfcd5acbd54512188e 100644
--- a/src/main/java/net/minecraft/world/item/ChorusFruitItem.java
+++ b/src/main/java/net/minecraft/world/item/ChorusFruitItem.java
@@ -9,6 +9,7 @@ import net.minecraft.world.entity.LivingEntity;
 import net.minecraft.world.entity.animal.Fox;
 import net.minecraft.world.entity.player.Player;
 import net.minecraft.world.level.Level;
+import org.bukkit.event.vehicle.VehicleExitEvent; // PaperDragon
 
 public class ChorusFruitItem extends Item {
 
@@ -31,6 +32,7 @@ public class ChorusFruitItem extends Item {
                 double d5 = user.getZ() + (user.getRandom().nextDouble() - 0.5D) * 16.0D;
 
                 if (user.isPassenger()) {
+                    user.dismountReason = VehicleExitEvent.DismountReason.TELEPORT; // PaperDragon
                     user.stopRiding();
                 }
 
