From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Thu, 18 Dec 2014 23:03:44 -0500
Subject: [PATCH] Add FlowerPotRemoveEvent


diff --git a/src/main/java/net/minecraft/world/level/block/FlowerPotBlock.java b/src/main/java/net/minecraft/world/level/block/FlowerPotBlock.java
index db05c1ea847d60ad45d33cd798cb34ad3f5cfd75..84c8bbe76997b6168447c7bad4845809a6555573 100644
--- a/src/main/java/net/minecraft/world/level/block/FlowerPotBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/FlowerPotBlock.java
@@ -80,6 +80,12 @@ public class FlowerPotBlock extends Block {
                 }
             } else {
                 ItemStack itemStack2 = new ItemStack(this.content);
+                // PaperDragon Start
+                com.domnian.paperdragon.events.FlowerPotRemoveEvent removeEvent = new com.domnian.paperdragon.events.FlowerPotRemoveEvent(
+                    player1, io.papermc.paper.util.MCUtil.toLocation(world, pos), itemStack2.getBukkitStack());
+                removeEvent.callEvent();
+                itemStack2 = ItemStack.fromBukkitCopy(removeEvent.getItem());
+                // PaperDragon End
                 if (itemStack.isEmpty()) {
                     player.setItemInHand(hand, itemStack2);
                 } else if (!player.addItem(itemStack2)) {
