From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Thu, 23 Jun 2016 22:30:39 -0400
Subject: [PATCH] Implement User Object API


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index ad15c4c20ae9e1858a2b914b0a1aa91ec1d7a012..78c4e88341eab3e2b57d6526cd7e1bda63a24fc4 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2563,4 +2563,17 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         return this.spigot;
     }
     // Spigot end
+
+    // PaperDragon start - User Object API
+    private final java.util.Map<org.bukkit.NamespacedKey, Object> userObjects = new java.util.HashMap<>();
+    public void setUserObject(@org.jetbrains.annotations.NotNull org.bukkit.NamespacedKey key, Object object) {
+        userObjects.put(key, object);
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T> T getUserObject(@org.jetbrains.annotations.NotNull org.bukkit.NamespacedKey key) {
+        return (T) userObjects.getOrDefault(key, null);
+    }
+    // PaperDragon end
+
 }
