From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Thu, 23 Jun 2016 22:30:39 -0400
Subject: [PATCH] Implement User Object API


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index d309d9f0b54e2a966ccfbb780359bb8e68d15ee3..fbf7616fd2f29575cf0147d641a902c89d62988a 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2669,4 +2669,17 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         return this.spigot;
     }
     // Spigot end
+
+    // PaperDragon start - User Object API
+    private final java.util.Map<org.bukkit.NamespacedKey, Object> userObjects = new java.util.HashMap<>();
+    public void setUserObject(@org.jetbrains.annotations.NotNull org.bukkit.NamespacedKey key, Object object) {
+        userObjects.put(key, object);
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T> T getUserObject(@org.jetbrains.annotations.NotNull org.bukkit.NamespacedKey key) {
+        return (T) userObjects.getOrDefault(key, null);
+    }
+    // PaperDragon end
+
 }
