From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Thu, 23 Jun 2016 22:30:39 -0400
Subject: [PATCH] Implement User Object API


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 76e2ae09855e0efaaa0856d2f49e4968adbccbdc..9cc47e74c1545354bf05d745d92d2d45a450e5e7 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2473,4 +2473,17 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         return this.spigot;
     }
     // Spigot end
+
+    // PaperDragon start - User Object API
+    private final java.util.Map<org.bukkit.NamespacedKey, Object> userObjects = new java.util.HashMap<>();
+    public void setUserObject(@org.jetbrains.annotations.NotNull org.bukkit.NamespacedKey key, Object object) {
+        userObjects.put(key, object);
+    }
+
+    @SuppressWarnings("unchecked")
+    public <T> T getUserObject(@org.jetbrains.annotations.NotNull org.bukkit.NamespacedKey key) {
+        return (T) userObjects.getOrDefault(key, null);
+    }
+    // PaperDragon end
+
 }
