From 1aedff71a7416df3e9fd6d1de7b069e818a162f3 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Thu, 23 Jun 2016 22:30:39 -0400
Subject: [PATCH] Implement User Object API

---
 .../org/bukkit/craftbukkit/entity/CraftPlayer.java   | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 7c18b22c7b..6cd63e586f 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2223,4 +2223,16 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         return spigot;
     }
     // Spigot end
+
+    // PaperDragon start - User Object API
+    private java.util.Map<org.bukkit.NamespacedKey, Object> userObjects = new java.util.HashMap<>();
+    public void setUserObject(org.bukkit.NamespacedKey key, Object object) {
+        userObjects.put(key, object);
+    }
+
+    public <T> T getUserObject(org.bukkit.NamespacedKey key) {
+        return (T) userObjects.getOrDefault(key, null);
+    }
+    // PaperDragon end
+
 }
-- 
2.20.1.windows.1

