From 18ba04c32e48b4295ae4ac9da4414234043cb9ed Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Thu, 23 Jun 2016 22:30:39 -0400
Subject: [PATCH] Implement User Object API

---
 .../org/bukkit/craftbukkit/entity/CraftPlayer.java   | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index b3aeb6870..0d2a1e7a0 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2093,4 +2093,16 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
         return spigot;
     }
     // Spigot end
+
+    // PaperDragon start - User Object API
+    private java.util.Map<Plugin, Object> userObjects = new java.util.HashMap<>();
+    public void setUserObject(Plugin owner, Object object) {
+        userObjects.put(owner, object);
+    }
+
+    public <T> T getUserObject(Plugin owner) {
+        return (T) userObjects.getOrDefault(owner, null);
+    }
+    // PaperDragon end
+
 }
-- 
2.17.1

