From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Wed, 4 Jul 2018 15:49:34 -0400
Subject: [PATCH] Fire PlayerKickEvent during server shutdown


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index e77e71b2c1b99aaa422272dd97374372b33bef91..c2ba6bdc4eb15bd25fa79aedf35859d6ff1cd2ab 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -441,11 +441,11 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
 
         PlayerKickEvent event = new PlayerKickEvent(this.player.getBukkitEntity(), reason, leaveMessage, cause); // Paper - Adventure & kick event reason
 
-        if (this.cserver.getServer().isRunning()) {
+        if (true) { // PaperDragon - don't care if its async either, fire it!
             this.cserver.getPluginManager().callEvent(event);
         }
 
-        if (event.isCancelled()) {
+        if (!server.hasStopped() && event.isCancelled()) { // PaperDragon
             // Do not kick the player
             return;
         }
