From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Mon, 19 Jun 2017 00:28:41 -0400
Subject: [PATCH] Add PaperDragon Command


diff --git a/src/main/java/com/domnian/paperdragon/DragonConfig.java b/src/main/java/com/domnian/paperdragon/DragonConfig.java
index 2f7bc8187f74099841a6572d35541bad8f60c5d9..e1ad836f928a1f337b461eda92d655556e030822 100644
--- a/src/main/java/com/domnian/paperdragon/DragonConfig.java
+++ b/src/main/java/com/domnian/paperdragon/DragonConfig.java
@@ -47,6 +47,7 @@ public class DragonConfig {
         config.options().copyDefaults(true);
 
         commands = new HashMap<>();
+        commands.put("paperdragon", new PaperDragonCommand("paperdragon"));
 
         version = getInt("config-version", 1);
         set("config-version", 1);
diff --git a/src/main/java/com/domnian/paperdragon/PaperDragonCommand.java b/src/main/java/com/domnian/paperdragon/PaperDragonCommand.java
new file mode 100644
index 0000000000000000000000000000000000000000..d704f68118e1fe2e2c4b992c154960e68d4b5281
--- /dev/null
+++ b/src/main/java/com/domnian/paperdragon/PaperDragonCommand.java
@@ -0,0 +1,38 @@
+package com.domnian.paperdragon;
+
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandSender;
+
+public class PaperDragonCommand extends Command {
+
+    public PaperDragonCommand(String name) {
+        super(name);
+        this.description = "PaperDragon-Related Commands";
+        this.usageMessage = "/paperdragon [metrics|stats]";
+        this.setPermission("bukkit.command.paperdragon");
+    }
+
+    @Override
+    public boolean execute(CommandSender sender, String commandLabel, String[] args) {
+        if (!testPermission(sender)) return true;
+
+        if (args.length != 1) {
+            sender.sendMessage(ChatColor.RED + "Usage: " + usageMessage);
+            return false;
+        }
+
+        switch (args[0]) {
+            case "metrics":
+            case "stats": {
+                sender.sendMessage(ChatColor.GREEN + "PaperDragon Metrics: " + ChatColor.AQUA + ChatColor.UNDERLINE + "https://bstats.org/plugin/server-implementation/PaperDragon");
+                break;
+            }
+            default: {
+                sender.sendMessage(ChatColor.RED + "Usage: " + usageMessage);
+            }
+        }
+
+        return true;
+    }
+}
