From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Thu, 15 Jun 2017 17:42:33 -0400
Subject: [PATCH] Settings for Default Advancement Broadcast Recievers


diff --git a/src/main/java/com/domnian/paperdragon/DragonConfig.java b/src/main/java/com/domnian/paperdragon/DragonConfig.java
index 86258b285cb8d39961a130618f1a9fed54dac889..c25990d237c3408917a69396efca8fc2aeb36c2c 100644
--- a/src/main/java/com/domnian/paperdragon/DragonConfig.java
+++ b/src/main/java/com/domnian/paperdragon/DragonConfig.java
@@ -160,6 +160,10 @@ public class DragonConfig {
         return config.getString(path, config.getString(path));
     }
 
+    public static boolean broadcastAdvancementsToServer = false;
+    private static void advancements() {
+        broadcastAdvancementsToServer = getBoolean("settings.broadcast-advancements-to-server", false);
+    }
     // TODO Add New Settings!! :D
 
 }
diff --git a/src/main/java/com/domnian/paperdragon/DragonWorldConfig.java b/src/main/java/com/domnian/paperdragon/DragonWorldConfig.java
index 71f698d648872341b7fef1326d45fdafd0694c32..f2fe260d8a0ac860b2af742735ade7ac70b0457a 100644
--- a/src/main/java/com/domnian/paperdragon/DragonWorldConfig.java
+++ b/src/main/java/com/domnian/paperdragon/DragonWorldConfig.java
@@ -73,6 +73,10 @@ public class DragonWorldConfig {
         return config.getString("world-settings." + worldName + "." + path, config.getString("world-settings.default." + path));
     }
 
+    public static boolean broadcastAdvancementsToWorld = false;
+    private void advancements() {
+        broadcastAdvancementsToWorld = getBoolean("world-settings.broadcast-advancements-to-world", false);
+    }
     // TODO Add New Settings!! :D
 
 }
diff --git a/src/main/java/net/minecraft/server/PlayerAdvancements.java b/src/main/java/net/minecraft/server/PlayerAdvancements.java
index 5f55931fa628bd2e5541fc1cf582e71608544555..acc86a040445118bf2fd892b10477671582ac396 100644
--- a/src/main/java/net/minecraft/server/PlayerAdvancements.java
+++ b/src/main/java/net/minecraft/server/PlayerAdvancements.java
@@ -307,6 +307,15 @@ public class PlayerAdvancements {
                     //this.playerList.broadcastMessage(io.papermc.paper.adventure.PaperAdventure.asVanilla(message), ChatType.SYSTEM, Util.NIL_UUID);
                     // Paper end
                     AdvancementBroadcastEvent broadcastEvent = new AdvancementBroadcastEvent(this.player.getBukkitEntity());
+                    if (com.domnian.paperdragon.DragonWorldConfig.broadcastAdvancementsToWorld) {
+                        List<net.minecraft.server.level.ServerPlayer> humans = this.player.getLevel().players;
+                        List<org.bukkit.entity.Player> players = new java.util.ArrayList<>();
+                        humans.forEach(human -> players.add(human.getBukkitEntity()));
+                        broadcastEvent.setReceivers(players);
+                    }
+                    if (com.domnian.paperdragon.DragonConfig.broadcastAdvancementsToServer) {
+                        broadcastEvent.setReceivers(new java.util.ArrayList<>(org.bukkit.Bukkit.getOnlinePlayers()));
+                    }
                     broadcastEvent.callEvent();
                     for (Player player : broadcastEvent.getReceivers()) {
                         ((CraftEntity) player).getHandle().sendMessage(io.papermc.paper.adventure.PaperAdventure.asVanilla(message), Util.NIL_UUID);
