From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 3 Jul 2016 19:16:19 -0400
Subject: [PATCH] Expose Spigot's BungeeCord Setting to API


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index afcfc2dcdc283b73113eb6fd8a290bad0fdb428b..8aad30d17d521ec1c5b216fe28b70985406c8b86 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -2685,4 +2685,11 @@ public final class CraftServer implements Server {
     }
 
     // Paper end
+
+    // PaperDragon start - Expose BungeeCord Setting
+    @Override
+    public boolean isBungeeEnabled() {
+        return org.spigotmc.SpigotConfig.bungee;
+    }
+    // PaperDragon end
 }
