From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Thu, 15 Jun 2017 17:42:33 -0400
Subject: [PATCH] Settings for Default Advancement Broadcast Recievers


diff --git a/src/main/java/com/domnian/paperdragon/DragonConfig.java b/src/main/java/com/domnian/paperdragon/DragonConfig.java
index 78f22ef4c08495e4b89f5a2eff4bdd3825d10d12..c033b4f3ef1aad4176686fcfd46a15f9655d7c07 100644
--- a/src/main/java/com/domnian/paperdragon/DragonConfig.java
+++ b/src/main/java/com/domnian/paperdragon/DragonConfig.java
@@ -160,6 +160,10 @@ public class DragonConfig {
         return config.getString(path, config.getString(path));
     }
 
+    public static boolean broadcastAdvancementsToServer = false;
+    private static void advancements() {
+        broadcastAdvancementsToServer = getBoolean("settings.broadcast-advancements-to-server", false);
+    }
     // TODO Add New Settings!! :D
 
 }
diff --git a/src/main/java/com/domnian/paperdragon/DragonWorldConfig.java b/src/main/java/com/domnian/paperdragon/DragonWorldConfig.java
index 69140c81740ba017888828ba163a4fbc2f7a6fa5..04de9955ba445d6d3a6ec638b510b1296ccab96a 100644
--- a/src/main/java/com/domnian/paperdragon/DragonWorldConfig.java
+++ b/src/main/java/com/domnian/paperdragon/DragonWorldConfig.java
@@ -73,6 +73,10 @@ public class DragonWorldConfig {
         return config.getString("world-settings." + worldName + "." + path, config.getString("world-settings.default." + path));
     }
 
+    public static boolean broadcastAdvancementsToWorld = false;
+    private void advancements() {
+        broadcastAdvancementsToWorld = getBoolean("world-settings.broadcast-advancements-to-world", false);
+    }
     // TODO Add New Settings!! :D
 
 }
diff --git a/src/main/java/net/minecraft/server/PlayerAdvancements.java b/src/main/java/net/minecraft/server/PlayerAdvancements.java
index 18b3283daafd3378cdb71c5979f6b96e6da46df7..9479c8fa254532504bad3ed25f621de590db2647 100644
--- a/src/main/java/net/minecraft/server/PlayerAdvancements.java
+++ b/src/main/java/net/minecraft/server/PlayerAdvancements.java
@@ -303,6 +303,15 @@ public class PlayerAdvancements {
                     //this.playerList.broadcastSystemMessage(io.papermc.paper.adventure.PaperAdventure.asVanilla(message), false);
                     // Paper end
                     AdvancementBroadcastEvent broadcastEvent = new AdvancementBroadcastEvent(this.player.getBukkitEntity());
+                    if (com.domnian.paperdragon.DragonWorldConfig.broadcastAdvancementsToWorld) {
+                        List<net.minecraft.server.level.ServerPlayer> humans = this.player.getLevel().players;
+                        List<org.bukkit.entity.Player> players = new java.util.ArrayList<>();
+                        humans.forEach(human -> players.add(human.getBukkitEntity()));
+                        broadcastEvent.setReceivers(players);
+                    }
+                    if (com.domnian.paperdragon.DragonConfig.broadcastAdvancementsToServer) {
+                        broadcastEvent.setReceivers(new java.util.ArrayList<>(org.bukkit.Bukkit.getOnlinePlayers()));
+                    }
                     broadcastEvent.callEvent();
                     for (Player player : broadcastEvent.getReceivers()) {
                         this.playerList.broadcastSystemMessage(io.papermc.paper.adventure.PaperAdventure.asVanilla(message), false);
