From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 3 Jul 2016 19:16:19 -0400
Subject: [PATCH] Implement Proxy Mode API


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 71a534660accecfc0e30e4d8b8b09fb484dc3802..4cab48599e9ab516fd37879c8380c85bdfbead19 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1,5 +1,6 @@
 package org.bukkit.craftbukkit;
 
+import com.domnian.paperdragon.ProxyMode; // PaperDragon
 import com.domnian.paperdragon.api.EntityTasksHandler; // PaperDragon
 import com.domnian.paperdragon.api.meta.MetaApiAccessor; // PaperDragon
 import com.google.common.base.Charsets;
@@ -2876,4 +2877,25 @@ public final class CraftServer implements Server {
     }
 
     // Paper end
+
+    // PaperDragon start
+    public ProxyMode proxyMode = ProxyMode.UNKNOWN;
+    @Override
+    public ProxyMode getProxyMode() {
+        if (proxyMode != ProxyMode.UNKNOWN) return proxyMode;
+        if (org.spigotmc.SpigotConfig.bungee) {
+            if (com.destroystokyo.paper.PaperConfig.bungeeOnlineMode) {
+                return proxyMode = ProxyMode.BUNGEECORD_ONLINE;
+            }
+            return proxyMode = ProxyMode.BUNGEECORD_OFFLINE;
+        } else if (com.destroystokyo.paper.PaperConfig.velocitySupport) {
+            if (com.destroystokyo.paper.PaperConfig.velocityOnlineMode) {
+                return proxyMode = ProxyMode.VELOCITY_ONLINE;
+            }
+            return proxyMode = ProxyMode.VELOCITY_OFFLINE;
+        }
+        return proxyMode = ProxyMode.NONE;
+    }
+    // PaperDragon end
+
 }
