From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 3 Jul 2016 19:16:19 -0400
Subject: [PATCH] Implement Proxy Mode API


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 7fd6f878e50c9820563a62e7d887d3f6fd8964ff..4d2c42c7fe339ed38ebb4e5cc4b858fa6606e0a7 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1,5 +1,6 @@
 package org.bukkit.craftbukkit;
 
+import com.domnian.paperdragon.ProxyMode; // PaperDragon
 import com.domnian.paperdragon.api.EntityTasksHandler; // PaperDragon
 import com.domnian.paperdragon.api.meta.MetaApiAccessor; // PaperDragon
 import com.google.common.base.Charsets;
@@ -18,6 +19,7 @@ import com.mojang.serialization.Lifecycle;
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.ByteBufOutputStream;
 import io.netty.buffer.Unpooled;
+import io.papermc.paper.configuration.GlobalConfiguration; // Paperdragon
 import it.unimi.dsi.fastutil.objects.Object2IntOpenHashMap;
 import java.awt.image.BufferedImage;
 import java.io.File;
@@ -237,6 +239,7 @@ import org.bukkit.scheduler.BukkitWorker;
 import org.bukkit.structure.StructureManager;
 import org.bukkit.util.StringUtil;
 import org.bukkit.util.permissions.DefaultPermissions;
+import org.jetbrains.annotations.NotNull; // PaperDragon
 import org.yaml.snakeyaml.Yaml;
 import org.yaml.snakeyaml.constructor.SafeConstructor;
 import org.yaml.snakeyaml.error.MarkedYAMLException;
@@ -2909,4 +2912,26 @@ public final class CraftServer implements Server {
     }
 
     // Paper end
+
+    // PaperDragon start
+    public ProxyMode proxyMode = ProxyMode.UNKNOWN;
+    @NotNull
+    @Override
+    public ProxyMode getProxyMode() {
+        if (proxyMode != ProxyMode.UNKNOWN) return proxyMode;
+        if (org.spigotmc.SpigotConfig.bungee) {
+            if (GlobalConfiguration.get().proxies.bungeeCord.onlineMode) {
+                return proxyMode = ProxyMode.BUNGEECORD_ONLINE;
+            }
+            return proxyMode = ProxyMode.BUNGEECORD_OFFLINE;
+        } else if (GlobalConfiguration.get().proxies.velocity.enabled) {
+            if (GlobalConfiguration.get().proxies.velocity.onlineMode) {
+                return proxyMode = ProxyMode.VELOCITY_ONLINE;
+            }
+            return proxyMode = ProxyMode.VELOCITY_OFFLINE;
+        }
+        return proxyMode = ProxyMode.NONE;
+    }
+    // PaperDragon end
+
 }
