From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 3 Jul 2016 19:16:19 -0400
Subject: [PATCH] Expose Spigot's BungeeCord Setting to API


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index f101adec6bf058cbec7a5c8bb1b1d938caa10c2c..ebbbc32b53a4e143b5b849ca7b5e46017c010942 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -2546,4 +2546,11 @@ public final class CraftServer implements Server {
     }
 
     // Paper end
+
+    // PaperDragon start - Expose BungeeCord Setting
+    @Override
+    public boolean isBungeeEnabled() {
+        return org.spigotmc.SpigotConfig.bungee;
+    }
+    // PaperDragon end
 }
