From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 7 Aug 2016 15:17:26 -0400
Subject: [PATCH] Implement PlayerHitPlayer Event

Fired when a player hits another player, regardless of gamemode

diff --git a/src/main/java/com/domnian/paperdragon/PDEventFactory.java b/src/main/java/com/domnian/paperdragon/PDEventFactory.java
new file mode 100644
index 0000000000000000000000000000000000000000..8e11aa0faa8083d7624d49e8f372e6063011ac22
--- /dev/null
+++ b/src/main/java/com/domnian/paperdragon/PDEventFactory.java
@@ -0,0 +1,20 @@
+package com.domnian.paperdragon;
+
+import com.domnian.paperdragon.events.PlayerHitPlayerEvent;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.entity.Entity;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+
+public class PDEventFactory {
+
+    public static void handlePlayerHitPlayer(Entity damager, Entity target) {
+        if ((damager instanceof ServerPlayer) && (target instanceof ServerPlayer)) {
+            Player from = (Player) damager.getBukkitEntity();
+            Player to = (Player) target.getBukkitEntity();
+            ItemStack item = from.getInventory().getItemInMainHand();
+            new PlayerHitPlayerEvent(from, to, item).callEvent();
+        }
+    }
+
+}
diff --git a/src/main/java/net/minecraft/world/entity/player/Player.java b/src/main/java/net/minecraft/world/entity/player/Player.java
index 7df610afc026d4229901ff541934b0bec33bd018..60f6268ae0f98ca9acc0b1be3409038ed51dc4d0 100644
--- a/src/main/java/net/minecraft/world/entity/player/Player.java
+++ b/src/main/java/net/minecraft/world/entity/player/Player.java
@@ -1254,6 +1254,7 @@ public abstract class Player extends LivingEntity {
     public void attack(Entity target) {
         if (target.isAttackable()) {
             if (!target.skipAttackInteraction(this)) {
+                com.domnian.paperdragon.PDEventFactory.handlePlayerHitPlayer(this, target); // PaperDragon - Implement PlayerHitPlayerEvent
                 float f = (float) this.getAttributeValue(Attributes.ATTACK_DAMAGE);
                 float f1;
 
