From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Fri, 22 Sep 2017 00:45:30 -0400
Subject: [PATCH] Add Method to Send a Mojangson (JSON) Message to a Player


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 3f1875fba3c118bb954bf98255555f5da9cfe526..dbd6f3e787ae574e7a2f194a5252804e90372627 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2948,6 +2948,11 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
     public <T> T getUserObject(@org.jetbrains.annotations.NotNull org.bukkit.NamespacedKey key) {
         return (T) userObjects.getOrDefault(key, null);
     }
+
+    public void sendJsonMessage(@NotNull String json) {
+        net.minecraft.network.chat.Component comp = net.minecraft.network.chat.Component.Serializer.fromJson(json);
+        if (comp != null) getHandle().sendSystemMessage(comp);
+    }
     // PaperDragon end
 
-}
+}
\ No newline at end of file
