From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Fri, 22 Sep 2017 00:45:30 -0400
Subject: [PATCH] Add Method to Send a Mojangson (JSON) Message to a Player


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index da03531c82a3bb1919f1b9532f9ab8c7da56297d..4f072664f67884df694853c51bd66773cf4bf8bc 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2800,6 +2800,11 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
     public <T> T getUserObject(@org.jetbrains.annotations.NotNull org.bukkit.NamespacedKey key) {
         return (T) userObjects.getOrDefault(key, null);
     }
+
+    public void sendJsonMessage(@NotNull String json) {
+        net.minecraft.network.chat.Component comp = net.minecraft.network.chat.Component.Serializer.fromJson(json);
+        if (comp != null) getHandle().sendSystemMessage(comp);
+    }
     // PaperDragon end
 
-}
+}
\ No newline at end of file
