From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Fri, 22 Sep 2017 00:45:30 -0400
Subject: [PATCH] Add Method to Send a Mojangson (JSON) Message to a Player


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index 9aa174dff0327a198fada101571d729dd7acee14..ee2ffd96865604106dd4c3648a8c4a1a17c79f9f 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2486,6 +2486,14 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
     public <T> T getUserObject(@org.jetbrains.annotations.NotNull org.bukkit.NamespacedKey key) {
         return (T) userObjects.getOrDefault(key, null);
     }
+
+    public void sendJsonMessage(String json) {
+        getHandle().sendMessage(
+                net.minecraft.network.chat.Component.Serializer.fromJson(json),
+                net.minecraft.network.chat.ChatType.SYSTEM,
+                Util.NIL_UUID
+        );
+    }
     // PaperDragon end
 
-}
+}
\ No newline at end of file
