From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Fri, 22 Sep 2017 00:45:30 -0400
Subject: [PATCH] Add Method to Send a Mojangson (JSON) Message to a Player


diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
index dc5522390b66f1180e375f6b411868744327aa97..275e47b939eb9410f459a06464b3607dea78ef44 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
@@ -2495,6 +2495,14 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
     public <T> T getUserObject(@org.jetbrains.annotations.NotNull org.bukkit.NamespacedKey key) {
         return (T) userObjects.getOrDefault(key, null);
     }
+
+    public void sendJsonMessage(String json) {
+        getHandle().sendMessage(
+                net.minecraft.network.chat.Component.Serializer.fromJson(json),
+                net.minecraft.network.chat.ChatType.SYSTEM,
+                Util.NIL_UUID
+        );
+    }
     // PaperDragon end
 
-}
+}
\ No newline at end of file
