From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 12 Jan 2020 19:50:51 -0500
Subject: [PATCH] Implement getLocation() for offline players


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index fe19b1f060cfed03b8cc57285ce393e0003d4242..5ea9b29b0ce85348f1448a72d0109e8f0642eaac 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -2190,6 +2190,7 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
                 nbt.putBoolean("Paper.FreezeLock", true);
             }
             // Paper end
+            nbt.putString("Dragon.DisconnectWorld", ((ServerLevel) this.level).serverLevelData.getLevelName()); // PaperDragon start - save disconnect world
             return nbt;
         } catch (Throwable throwable) {
             CrashReport crashreport = CrashReport.forThrowable(throwable, "Saving entity NBT");
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java b/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
index 1f2bc88d4570c6ef00e67a772b745e0b0c98e051..47c9e7c9de620a5c55ca90314f7866c632a24842 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
@@ -528,4 +528,32 @@ public class CraftOfflinePlayer implements OfflinePlayer, ConfigurationSerializa
             manager.save();
         }
     }
+
+    // PaperDragon start
+    @Override
+    @org.jetbrains.annotations.NotNull
+    public Location getLocation() {
+        if (this.getPlayer() != null) return getPlayer().getLocation();
+        CompoundTag data = getData();
+        net.minecraft.nbt.ListTag pos = data.getList("Pos", 6); // 6 = NBTTagDouble
+        net.minecraft.nbt.ListTag rot = data.getList("Rotation", 5); // 5 = NBTTagFloat
+
+        UUID worldUUID = new UUID(data.getLong("WorldUUIDMost"), data.getLong("WorldUUIDLeast"));
+        org.bukkit.World world = org.bukkit.Bukkit.getWorld(worldUUID);
+        if (world == null) {
+            if (data.contains("Dragon.DisconnectWorld", 8)) {
+                world = Bukkit.getWorld(data.getString("Dragon.DisconnectWorld"));
+            } else {
+                world = Bukkit.getWorld(server.getHandle().getServer().getWorldData().getLevelName());
+            }
+        }
+
+        return new Location(
+                world, // World
+                pos.getDouble(0), pos.getDouble(1), pos.getDouble(2), // X, Y, Z
+                ((net.minecraft.nbt.FloatTag) rot.get(0)).getAsFloat(), // Yaw
+                ((net.minecraft.nbt.FloatTag) rot.get(1)).getAsFloat() // Pitch
+        );
+    }
+    // PaperDragon end
 }
