From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 12 Jan 2020 19:50:51 -0500
Subject: [PATCH] Implement getLocation() for offline players


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index 794c9410280745472f40e9958cf882883db1f49f..8b9506205664bb466dc0c0616d3aeb08c3a5b0df 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -2276,6 +2276,7 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource {
                 nbt.putBoolean("Paper.FreezeLock", true);
             }
             // Paper end
+            nbt.putString("Dragon.DisconnectWorld", ((ServerLevel) this.level).serverLevelData.getLevelName()); // PaperDragon start - save disconnect world
             return nbt;
         } catch (Throwable throwable) {
             CrashReport crashreport = CrashReport.forThrowable(throwable, "Saving entity NBT");
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java b/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
index 6d2ba650f53de8a460857f1846401a20b50cc43c..e4749b0aeb8d9ad020b704a1bb2e9ca9085857f7 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
@@ -549,4 +549,32 @@ public class CraftOfflinePlayer implements OfflinePlayer, ConfigurationSerializa
             manager.save();
         }
     }
+
+    // PaperDragon start
+    @Override
+    @org.jetbrains.annotations.NotNull
+    public Location getLocation() {
+        if (this.getPlayer() != null) return getPlayer().getLocation();
+        CompoundTag data = getData();
+        net.minecraft.nbt.ListTag pos = data.getList("Pos", 6); // 6 = NBTTagDouble
+        net.minecraft.nbt.ListTag rot = data.getList("Rotation", 5); // 5 = NBTTagFloat
+
+        UUID worldUUID = new UUID(data.getLong("WorldUUIDMost"), data.getLong("WorldUUIDLeast"));
+        org.bukkit.World world = org.bukkit.Bukkit.getWorld(worldUUID);
+        if (world == null) {
+            if (data.contains("Dragon.DisconnectWorld", 8)) {
+                world = Bukkit.getWorld(data.getString("Dragon.DisconnectWorld"));
+            } else {
+                world = Bukkit.getWorld(server.getHandle().getServer().getWorldData().getLevelName());
+            }
+        }
+
+        return new Location(
+                world, // World
+                pos.getDouble(0), pos.getDouble(1), pos.getDouble(2), // X, Y, Z
+                ((net.minecraft.nbt.FloatTag) rot.get(0)).getAsFloat(), // Yaw
+                ((net.minecraft.nbt.FloatTag) rot.get(1)).getAsFloat() // Pitch
+        );
+    }
+    // PaperDragon end
 }
