From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 12 Jan 2020 19:50:51 -0500
Subject: [PATCH] Implement getLocation() for offline players


diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
index a9e8507af88d65971b35b0b4bb2d88efa92d99ba..9e8d36ef91bad6af2b98c5f299bdad398920ff27 100644
--- a/src/main/java/net/minecraft/world/entity/Entity.java
+++ b/src/main/java/net/minecraft/world/entity/Entity.java
@@ -2171,6 +2171,7 @@ public abstract class Entity implements Nameable, EntityAccess, CommandSource, i
                 nbt.putBoolean("Paper.FromNetherPortal", true);
             }
             // Paper end
+            nbt.putString("Dragon.DisconnectWorld", ((ServerLevel) this.level).serverLevelData.getLevelName()); // PaperDragon start - save disconnect world
             return nbt;
         } catch (Throwable throwable) {
             CrashReport crashreport = CrashReport.forThrowable(throwable, "Saving entity NBT");
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java b/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
index 88bc0807e8bf66a65422f85f1112336334eb3de2..76ed77f71c2c008a8dc2045ee52f8c5661bce0cd 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
@@ -23,6 +23,7 @@ import org.bukkit.entity.EntityType;
 import org.bukkit.entity.Player;
 import org.bukkit.metadata.MetadataValue;
 import org.bukkit.plugin.Plugin;
+import org.jetbrains.annotations.NotNull;
 
 @SerializableAs("Player")
 public class CraftOfflinePlayer implements OfflinePlayer, ConfigurationSerializable {
@@ -525,4 +526,32 @@ public class CraftOfflinePlayer implements OfflinePlayer, ConfigurationSerializa
             manager.save();
         }
     }
+
+    // PaperDragon start
+    @NotNull
+    @Override
+    public Location getLocation() {
+        if (this.getPlayer() != null) return getPlayer().getLocation();
+        CompoundTag data = getData();
+        net.minecraft.nbt.ListTag pos = data.getList("Pos", 6); // 6 = NBTTagDouble
+        net.minecraft.nbt.ListTag rot = data.getList("Rotation", 5); // 5 = NBTTagFloat
+
+        UUID worldUUID = new UUID(data.getLong("WorldUUIDMost"), data.getLong("WorldUUIDLeast"));
+        org.bukkit.World world = org.bukkit.Bukkit.getWorld(worldUUID);
+        if (world == null) {
+            if (data.contains("Dragon.DisconnectWorld", 8)) {
+                world = Bukkit.getWorld(data.getString("Dragon.DisconnectWorld"));
+            } else {
+                world = Bukkit.getWorld(server.getHandle().getServer().getWorldData().getLevelName());
+            }
+        }
+
+        return new Location(
+                world, // World
+                pos.getDouble(0), pos.getDouble(1), pos.getDouble(2), // X, Y, Z
+                ((net.minecraft.nbt.FloatTag) rot.get(0)).getAsFloat(), // Yaw
+                ((net.minecraft.nbt.FloatTag) rot.get(1)).getAsFloat() // Pitch
+        );
+    }
+    // PaperDragon end
 }
