From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Fri, 17 Jan 2020 21:32:10 -0600
Subject: [PATCH] Add the BlockHarvestBeehiveEvent


diff --git a/src/main/java/net/minecraft/core/dispenser/DispenseItemBehavior.java b/src/main/java/net/minecraft/core/dispenser/DispenseItemBehavior.java
index 0d34026a70c72661a9ba6b319690370e589714cc..71a2c2181fa09b94b8459d2fb8e77edf8f76a0cb 100644
--- a/src/main/java/net/minecraft/core/dispenser/DispenseItemBehavior.java
+++ b/src/main/java/net/minecraft/core/dispenser/DispenseItemBehavior.java
@@ -974,6 +974,11 @@ public interface DispenseItemBehavior {
                 if (iblockdata.is((Tag) BlockTags.BEEHIVES, (blockbase_blockdata) -> {
                     return blockbase_blockdata.hasProperty(BeehiveBlock.HONEY_LEVEL);
                 }) && (Integer) iblockdata.getValue(BeehiveBlock.HONEY_LEVEL) >= 5) {
+                    // PaperDragon start
+                    if (!(new com.domnian.paperdragon.events.BlockHarvestBeehiveEvent(bukkitBlock, net.minecraft.server.MCUtil.toBukkitBlock(worldserver, blockposition), craftItem).callEvent())) {
+                        return stack;
+                    }
+                    // PaperDragon end
                     ((BeehiveBlock) iblockdata.getBlock()).releaseBeesAndResetHoneyLevel((Level) worldserver, iblockdata, blockposition, (Player) null, BeehiveBlockEntity.BeeReleaseStatus.BEE_RELEASED);
                     this.setSuccess(true);
                     return this.a(pointer, stack, new ItemStack(Items.HONEY_BOTTLE));
