From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Mon, 27 Jan 2020 00:54:51 -0500
Subject: [PATCH] Add Method to Save Offline Player Data to File

WARNING: This is a dangerous operation!

diff --git a/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java b/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
index 0b49d9099afe9219134b338d78caf1a08a70c746..a9609d29736ce96c1b83d703f0aaeb4c0bd958e4 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
@@ -565,5 +565,22 @@ public class CraftOfflinePlayer implements OfflinePlayer, ConfigurationSerializa
                 ((net.minecraft.nbt.FloatTag) rot.get(1)).getAsFloat() // Pitch
         );
     }
+
+    /** @deprecated Internal use only - <strong>THIS IS A DANGEROUS OPERATION</strong> */
+    @Deprecated
+    private boolean saveData(CompoundTag data) {
+        if (isOnline()) throw new UnsupportedOperationException("Dangerous Operation - Trying to save data of player while online!");
+        try {
+            File file = new File(storage.getPlayerDir(), getUniqueId() + ".dat.tmp");
+            File dataFile = getDataFile();
+
+            net.minecraft.nbt.NbtIo.writeCompressed(data, new java.io.FileOutputStream(file));
+            if (dataFile.exists()) dataFile.delete();
+            return file.renameTo(dataFile);
+        } catch (Exception exception) {
+            exception.printStackTrace();
+            return false;
+        }
+    }
     // PaperDragon end
 }
