From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Mon, 27 Jan 2020 00:56:50 -0500
Subject: [PATCH] Implement teleport() for offline players


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java b/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
index a9609d29736ce96c1b83d703f0aaeb4c0bd958e4..1148c95fe54748ce92b6173b45a36d445dae859a 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftOfflinePlayer.java
@@ -582,5 +582,33 @@ public class CraftOfflinePlayer implements OfflinePlayer, ConfigurationSerializa
             return false;
         }
     }
+
+    @Override
+    public boolean teleport(Location loc) {
+        if (isOnline()) {
+            getPlayer().teleport(loc);
+            return true;
+        }
+        CompoundTag data = getData();
+
+        net.minecraft.nbt.ListTag rot = new net.minecraft.nbt.ListTag();
+        rot.add(0, net.minecraft.nbt.FloatTag.valueOf(loc.getYaw()));
+        rot.add(1, net.minecraft.nbt.FloatTag.valueOf(loc.getPitch()));
+        data.put("Rotation", rot);
+
+        net.minecraft.nbt.ListTag pos = new net.minecraft.nbt.ListTag();
+        pos.add(0, net.minecraft.nbt.DoubleTag.valueOf(loc.getX()));
+        pos.add(1, net.minecraft.nbt.DoubleTag.valueOf(loc.getY()));
+        pos.add(2, net.minecraft.nbt.DoubleTag.valueOf(loc.getZ()));
+        data.put("Pos", pos);
+
+        UUID worldUUID = loc.getWorld().getUID();
+        data.put("WorldUUIDMost",  net.minecraft.nbt.LongTag.valueOf(worldUUID.getMostSignificantBits()));
+        data.put("WorldUUIDLeast", net.minecraft.nbt.LongTag.valueOf(worldUUID.getLeastSignificantBits()));
+
+        data.putString("Dragon.DisconnectWorld", loc.getWorld().getName());
+
+        return saveData(data);
+    }
     // PaperDragon end
 }
