From 08bf1209a58286f92a5e6c919d926f1aacff93c1 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Wed, 1 Jul 2020 23:10:08 -0400
Subject: [PATCH] Allow using components to set display name

---
 .../org/bukkit/craftbukkit/inventory/CraftMetaItem.java    | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaItem.java b/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaItem.java
index dcdce99752..4fe65993da 100644
--- a/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaItem.java
+++ b/src/main/java/org/bukkit/craftbukkit/inventory/CraftMetaItem.java
@@ -888,6 +888,13 @@ class CraftMetaItem implements ItemMeta, Damageable, Repairable, BlockDataMeta {
         this.displayName = CraftChatMessage.fromStringOrNull(name);
     }
 
+    // PaperDragon start - Allow using components to set display name
+    @Override
+    public final void setDisplayName(net.md_5.bungee.api.chat.BaseComponent... name) {
+        this.displayName = IChatBaseComponent.ChatSerializer.jsonToComponent(net.md_5.bungee.chat.ComponentSerializer.toString(name));
+    }
+    // PaperDragon end
+
     @Override
     public boolean hasDisplayName() {
         return displayName != null;
-- 
2.20.1.windows.1

