From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Thu, 6 Nov 2014 21:36:14 -0500
Subject: [PATCH] Mark Withers as Persistent

We don't want them despawning due to player range

diff --git a/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java b/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
index b9d42e59a9f9e9ee4e6a06e2245a00dcc34f9002..fe2df5ea9c65dd801e83e6d849de0d83ba1582b5 100644
--- a/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
+++ b/src/main/java/net/minecraft/world/entity/boss/wither/WitherBoss.java
@@ -594,6 +594,7 @@ public class WitherBoss extends Monster implements PowerableMob, RangedAttackMob
         return this.getHealth() <= this.getMaxHealth() / 2.0F;
     }
 
+    @Override public boolean removeWhenFarAway(double distanceSquared) { return false; } // PaperDragon
     @Override
     public MobType getMobType() {
         return MobType.UNDEAD;
