From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Mon, 5 Jul 2021 22:01:39 -0500
Subject: [PATCH] DisperWaxBlockEvent


diff --git a/src/main/java/net/minecraft/core/dispenser/DispenseItemBehavior.java b/src/main/java/net/minecraft/core/dispenser/DispenseItemBehavior.java
index fcafaaf024bb6541fcf66fc6c3c0bbf9fab8c656..82f51acd4375bad90e5b24918c2f8ed9d73b9b56 100644
--- a/src/main/java/net/minecraft/core/dispenser/DispenseItemBehavior.java
+++ b/src/main/java/net/minecraft/core/dispenser/DispenseItemBehavior.java
@@ -1111,6 +1111,13 @@ public interface DispenseItemBehavior {
                 Optional<BlockState> optional = HoneycombItem.getWaxed(iblockdata);
 
                 if (optional.isPresent()) {
+                    // PaperDragon start
+                    org.bukkit.block.Block block = worldserver.getWorld().getBlockAt(pointer.getPos().getX(), pointer.getPos().getY(), pointer.getPos().getZ());
+                    CraftItemStack craftItem = CraftItemStack.asCraftMirror(stack);
+                    if (!(new com.domnian.paperdragon.events.DispenserWaxBlockEvent(net.minecraft.server.MCUtil.toBukkitBlock(worldserver, blockposition), block, craftItem).callEvent())) {
+                        return stack;
+                    }
+                    // PaperDragon end
                     worldserver.setBlockAndUpdate(blockposition, (BlockState) optional.get());
                     worldserver.levelEvent(3003, blockposition, 0);
                     stack.shrink(1);
