From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Mon, 5 Jul 2021 22:48:17 -0500
Subject: [PATCH] Add PlayerCandleCakeEvent


diff --git a/src/main/java/net/minecraft/world/level/block/CakeBlock.java b/src/main/java/net/minecraft/world/level/block/CakeBlock.java
index 1a87a1553bb94e33c86b9d8947cf39fd193e7859..5f1f490a2f7cea17a21f5afca1b2a1990a077fce 100644
--- a/src/main/java/net/minecraft/world/level/block/CakeBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/CakeBlock.java
@@ -55,6 +55,7 @@ public class CakeBlock extends Block {
             Block block = Block.byItem(item);
 
             if (block instanceof CandleBlock) {
+                new com.domnian.paperdragon.events.PlayerCandleCakeEvent((org.bukkit.entity.Player) player.getBukkitEntity(), net.minecraft.server.MCUtil.toLocation(world, pos), itemstack.getBukkitStack()).callEvent(); // PaperDragon
                 if (!player.isCreative()) {
                     itemstack.shrink(1);
                 }
