From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Tue, 3 Aug 2021 10:39:44 -0500
Subject: [PATCH] Do not process chat/commands before player has joined


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index c57c6709fbd227f45c0ac74644d0ec3819eb6b82..6d6f83711cbe492913eb87872eadc7b39576e109 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -2211,6 +2211,7 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Ser
 
             this.send(new ClientboundSystemChatPacket(Component.translatable("chat.disabled.options").withStyle(ChatFormatting.RED), i));
             return false;
+        } else if (!player.didPlayerJoinEvent) { return false; // PaperDragon - do not handle chat messages before they joined
         } else {
             this.player.resetLastActionTime();
             return true;
