From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Sat, 10 Jul 2021 23:31:59 -0500
Subject: [PATCH] Implement VegetationPatchPlaceEvent


diff --git a/src/main/java/net/minecraft/world/level/levelgen/feature/VegetationPatchFeature.java b/src/main/java/net/minecraft/world/level/levelgen/feature/VegetationPatchFeature.java
index 89732a41752eda3c2ccc07e9580ded8266f8d4f0..f0271fd4a58642ab2065f8136efdaae1b9fbc015 100644
--- a/src/main/java/net/minecraft/world/level/levelgen/feature/VegetationPatchFeature.java
+++ b/src/main/java/net/minecraft/world/level/levelgen/feature/VegetationPatchFeature.java
@@ -67,6 +67,15 @@ public class VegetationPatchFeature extends Feature<VegetationPatchConfiguration
                     if (world.isEmptyBlock(mutableBlockPos) && blockState.isFaceSturdy(world, mutableBlockPos2, config.surface.getDirection().getOpposite())) {
                         int l = config.depth.sample(random) + (config.extraBottomBlockChance > 0.0F && random.nextFloat() < config.extraBottomBlockChance ? 1 : 0);
                         BlockPos blockPos = mutableBlockPos2.immutable();
+                        // PaperDragon start
+                        if (org.bukkit.Bukkit.isPrimaryThread()) {
+                            org.bukkit.block.Block origin = net.minecraft.server.MCUtil.toBukkitBlock(world.getLevel(), pos);
+                            org.bukkit.block.Block place = net.minecraft.server.MCUtil.toBukkitBlock(world.getLevel(), blockPos);
+                            if (!(new com.domnian.paperdragon.events.VegetationPatchPlaceEvent(place, origin).callEvent())) {
+                                continue;
+                            }
+                        }
+                        // PaperDragon end
                         boolean bl6 = this.placeGround(world, config, replaceable, random, mutableBlockPos2, l);
                         if (bl6) {
                             set.add(blockPos);
@@ -82,6 +91,15 @@ public class VegetationPatchFeature extends Feature<VegetationPatchConfiguration
     protected void distributeVegetation(FeaturePlaceContext<VegetationPatchConfiguration> context, WorldGenLevel world, VegetationPatchConfiguration config, Random random, Set<BlockPos> positions, int radiusX, int radiusZ) {
         for(BlockPos blockPos : positions) {
             if (config.vegetationChance > 0.0F && random.nextFloat() < config.vegetationChance) {
+                // PaperDragon start
+                if (org.bukkit.Bukkit.isPrimaryThread()) {
+                    org.bukkit.block.Block origin = net.minecraft.server.MCUtil.toBukkitBlock(context.level().getLevel(), context.origin());
+                    org.bukkit.block.Block place = net.minecraft.server.MCUtil.toBukkitBlock(context.level().getLevel(), blockPos);
+                    if (!(new com.domnian.paperdragon.events.VegetationPatchPlaceEvent(place, origin).callEvent())) {
+                        continue;
+                    }
+                }
+                // PaperDragon end
                 this.placeVegetation(world, config, context.chunkGenerator(), random, blockPos);
             }
         }
