From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: chickeneer <emcchickeneer@gmail.com>
Date: Mon, 5 Jul 2021 22:48:17 -0500
Subject: [PATCH] Add PlayerCandleCakeEvent


diff --git a/src/main/java/net/minecraft/world/level/block/CakeBlock.java b/src/main/java/net/minecraft/world/level/block/CakeBlock.java
index 397b8945bd0c2f413b17ba95cfb31befca252595..863e86dd261c2d5e78e21d114448928c7f79d259 100644
--- a/src/main/java/net/minecraft/world/level/block/CakeBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/CakeBlock.java
@@ -55,6 +55,7 @@ public class CakeBlock extends Block {
             Block block = Block.byItem(item);
 
             if (block instanceof CandleBlock) {
+                new com.domnian.paperdragon.events.PlayerCandleCakeEvent((org.bukkit.entity.Player) player.getBukkitEntity(), io.papermc.paper.util.MCUtil.toLocation(world, pos), itemstack.getBukkitStack()).callEvent(); // PaperDragon
                 if (!player.isCreative()) {
                     itemstack.shrink(1);
                 }
