From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 11 Jul 2021 20:28:57 -0400
Subject: [PATCH] Add option to force 2D search for nether portals


diff --git a/src/main/java/com/domnian/paperdragon/DragonWorldConfig.java b/src/main/java/com/domnian/paperdragon/DragonWorldConfig.java
index 107135de9e1b06074f69c84d654a6b22ee8a1c67..a153f90efd1ae27f5fb84381fe57df83ed6b8bab 100644
--- a/src/main/java/com/domnian/paperdragon/DragonWorldConfig.java
+++ b/src/main/java/com/domnian/paperdragon/DragonWorldConfig.java
@@ -78,6 +78,11 @@ public class DragonWorldConfig {
     private void advancements() {
         broadcastAdvancementsToWorld = getBoolean("world-settings.broadcast-advancements-to-world", false);
     }
+
+    public boolean force2DPortalSearch = false;
+    private void portalOverrides() {
+        force2DPortalSearch = getBoolean("force-2d-portal-search", false);
+    }
     // TODO Add New Settings!! :D
 
 }
diff --git a/src/main/java/net/minecraft/world/level/portal/PortalForcer.java b/src/main/java/net/minecraft/world/level/portal/PortalForcer.java
index d5ba2e679ed1858ea18e18feffce50544ae036c2..aca639e26557ad4daf4db9deb6202cf4af16d3d8 100644
--- a/src/main/java/net/minecraft/world/level/portal/PortalForcer.java
+++ b/src/main/java/net/minecraft/world/level/portal/PortalForcer.java
@@ -56,6 +56,11 @@ public class PortalForcer {
         Optional<PoiRecord> optional = villageplace.getInSquare((villageplacetype) -> {
             return villageplacetype == PoiType.NETHER_PORTAL;
         }, blockposition, i, PoiManager.Occupancy.ANY).sorted(Comparator.comparingDouble((PoiRecord villageplacerecord) -> { // CraftBukkit - decompile error
+            // PaperDragon start - allow forcing the portal search to use a 2D search instead of 3D
+            if (level.dragonConfig.force2DPortalSearch) {
+                return villageplacerecord.getPos().atY(0).distSqr(blockposition.atY(0));
+            }
+            // PaperDragon end
             return villageplacerecord.getPos().distSqr(blockposition);
         }).thenComparingInt((villageplacerecord) -> {
             return villageplacerecord.getPos().getY();
