From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 3 Apr 2022 09:37:54 -0400
Subject: [PATCH] Add setting to disable 'Improve Hopper Performance' patch


diff --git a/src/main/java/com/domnian/paperdragon/DragonWorldConfig.java b/src/main/java/com/domnian/paperdragon/DragonWorldConfig.java
index f2fe260d8a0ac860b2af742735ade7ac70b0457a..b697e2b0b9540b7cb2a9d080478591153638a45b 100644
--- a/src/main/java/com/domnian/paperdragon/DragonWorldConfig.java
+++ b/src/main/java/com/domnian/paperdragon/DragonWorldConfig.java
@@ -77,6 +77,10 @@ public class DragonWorldConfig {
     private void advancements() {
         broadcastAdvancementsToWorld = getBoolean("world-settings.broadcast-advancements-to-world", false);
     }
-    // TODO Add New Settings!! :D
+
+    public boolean disableImprovedHopperPerformance = false;
+    private void hoppers() {
+        disableImprovedHopperPerformance = getBoolean("disable-improved-hopper-performance", false);
+    }
 
 }
diff --git a/src/main/java/net/minecraft/world/level/block/entity/HopperBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/HopperBlockEntity.java
index 12b92d90d344f9af7a9997fbb0cfb91c461c035b..ff20b2f9105578e1c24cede245f5144919c47505 100644
--- a/src/main/java/net/minecraft/world/level/block/entity/HopperBlockEntity.java
+++ b/src/main/java/net/minecraft/world/level/block/entity/HopperBlockEntity.java
@@ -487,7 +487,7 @@ public class HopperBlockEntity extends RandomizableContainerBlockEntity implemen
                 }
                 // Paper end
             });
-        } else if (shouldSuckIn(hopper)) { // PaperDragon
+        } else if (shouldSuckIn(hopper) || world.dragonConfig.disableImprovedHopperPerformance) { // PaperDragon
             Iterator iterator = HopperBlockEntity.getItemsAtAndAbove(world, hopper).iterator();
 
             ItemEntity entityitem;
