From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 2 Jan 2022 10:10:48 -0500
Subject: [PATCH] Add methods to get a raid's id and to cancel it


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftRaid.java b/src/main/java/org/bukkit/craftbukkit/CraftRaid.java
index b29620edac4e9a7a57fe5d8c75e6b78aebeafa0a..52f526be35347aacca3f72808c35faca37f0bccf 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftRaid.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftRaid.java
@@ -85,6 +85,15 @@ public final class CraftRaid implements Raid {
         Preconditions.checkArgument(1 <= total && total <= 7, "Total waves out of bounds 1 <= %s <= 7", total);
         handle.numGroups = total;
     }
+    @Override
+    public int getId() {
+        return handle.getId();
+    }
+    @Override
+    public void cancel() {
+        if (handle.isStopped()) return;
+        handle.stop();
+    }
     // PaperDragon end
     @Override
     public float getTotalHealth() {
