From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 2 Jan 2022 10:10:48 -0500
Subject: [PATCH] Add methods to get a raid's id and to cancel it


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftRaid.java b/src/main/java/org/bukkit/craftbukkit/CraftRaid.java
index 14c59db2dd09996779e4c231150e2eb35be2a50e..0195230c1ca8a69e3663af4b50dc252238384436 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftRaid.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftRaid.java
@@ -86,6 +86,15 @@ public final class CraftRaid implements Raid {
         Preconditions.checkArgument(1 <= total && total <= 7, "Total waves out of bounds 1 <= %s <= 7", total);
         handle.numGroups = total;
     }
+    @Override
+    public int getId() {
+        return handle.getId();
+    }
+    @Override
+    public void cancel() {
+        if (handle.isStopped()) return;
+        handle.stop();
+    }
     // PaperDragon end
     @Override
     public float getTotalHealth() {
