From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sun, 2 Oct 2022 21:23:17 -0400
Subject: [PATCH] Implement Allay Change Item Event


diff --git a/src/main/java/net/minecraft/world/entity/animal/allay/Allay.java b/src/main/java/net/minecraft/world/entity/animal/allay/Allay.java
index 2d87b740e84435ef9bd3eee3d3e2af9f707a0826..abf6df94c1d453c907d4dba3663e9eaf39d73ad7 100644
--- a/src/main/java/net/minecraft/world/entity/animal/allay/Allay.java
+++ b/src/main/java/net/minecraft/world/entity/animal/allay/Allay.java
@@ -68,6 +68,7 @@ import net.minecraft.world.level.gameevent.PositionSource;
 import net.minecraft.world.level.gameevent.vibrations.VibrationListener;
 import net.minecraft.world.phys.AABB;
 import net.minecraft.world.phys.Vec3;
+import org.bukkit.craftbukkit.inventory.CraftItemStack;
 import org.jetbrains.annotations.Nullable;
 import org.slf4j.Logger;
 
@@ -324,12 +325,36 @@ public class Allay extends PathfinderMob implements InventoryCarrier {
             ItemStack itemstack2 = itemstack.copy();
 
             itemstack2.setCount(1);
+            // PaperDragon start - Allay events
+            if (com.domnian.paperdragon.events.AllayChangeItemEvent.getHandlerList().hasListeners()) {
+                com.domnian.paperdragon.events.AllayChangeItemEvent event = new com.domnian.paperdragon.events.AllayChangeItemEvent(
+                        (org.bukkit.entity.Allay) this.getBukkitEntity(),
+                        (org.bukkit.entity.Player) player.getBukkitEntity(),
+                        com.domnian.paperdragon.events.AllayChangeItemEvent.Action.GIVE,
+                        org.bukkit.craftbukkit.inventory.CraftItemStack.asCraftMirror(itemstack2)
+                );
+                event.callEvent();
+                itemstack2 = org.bukkit.craftbukkit.inventory.CraftItemStack.asNMSCopy(event.getItem());
+            }
+            // PaperDragon end
             this.setItemInHand(InteractionHand.MAIN_HAND, itemstack2);
             this.removeInteractionItem(player, itemstack);
             this.level.playSound(player, (Entity) this, SoundEvents.ALLAY_ITEM_GIVEN, SoundSource.NEUTRAL, 2.0F, 1.0F);
             this.getBrain().setMemory(MemoryModuleType.LIKED_PLAYER, player.getUUID()); // CraftBukkit - decompile error
             return InteractionResult.SUCCESS;
         } else if (!itemstack1.isEmpty() && hand == InteractionHand.MAIN_HAND && itemstack.isEmpty()) {
+            // PaperDragon start - Allay events
+            if (com.domnian.paperdragon.events.AllayChangeItemEvent.getHandlerList().hasListeners()) {
+                com.domnian.paperdragon.events.AllayChangeItemEvent event = new com.domnian.paperdragon.events.AllayChangeItemEvent(
+                        (org.bukkit.entity.Allay) this.getBukkitEntity(),
+                        (org.bukkit.entity.Player) player.getBukkitEntity(),
+                        com.domnian.paperdragon.events.AllayChangeItemEvent.Action.TAKE,
+                        org.bukkit.craftbukkit.inventory.CraftItemStack.asCraftMirror(itemstack1)
+                );
+                event.callEvent();
+                itemstack1 = org.bukkit.craftbukkit.inventory.CraftItemStack.asNMSCopy(event.getItem());
+            }
+            // PaperDragon end
             this.setItemSlot(EquipmentSlot.MAINHAND, ItemStack.EMPTY);
             this.level.playSound(player, (Entity) this, SoundEvents.ALLAY_ITEM_TAKEN, SoundSource.NEUTRAL, 2.0F, 1.0F);
             this.swing(InteractionHand.MAIN_HAND);
