From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: willies952002 <admin@domnian.com>
Date: Sat, 30 Jul 2022 22:50:38 -0400
Subject: [PATCH] Add Adventure support to Chat Preview Event


diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index 3272befc1adde04fc77f176c3631207830af77ce..d0bd933ea597aeadd08cf5e14989f70d364009e0 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -2609,15 +2609,16 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
             }
             // Paper end
             return CompletableFuture.supplyAsync(() -> {
-                AsyncPlayerChatPreviewEvent event = new AsyncPlayerChatPreviewEvent(true, entityplayer.getBukkitEntity(), CraftChatMessage.fromComponent(ichatbasecomponent), new LazyPlayerSet(this));
+                net.kyori.adventure.text.Component originalComponent = io.papermc.paper.adventure.PaperAdventure.asAdventure(ichatbasecomponent); // PaperDragon
+                AsyncPlayerChatPreviewEvent event = new AsyncPlayerChatPreviewEvent(true, entityplayer.getBukkitEntity(), CraftChatMessage.fromComponent(ichatbasecomponent), new LazyPlayerSet(this), originalComponent); // PaperDragon - pass Adventure Component to preview event
                 String originalFormat = event.getFormat(), originalMessage = event.getMessage();
                 this.server.getPluginManager().callEvent(event);
 
-                if (originalFormat.equals(event.getFormat()) && originalMessage.equals(event.getMessage()) && event.getPlayer().getName().equalsIgnoreCase(event.getPlayer().getDisplayName())) {
+                if (originalFormat.equals(event.getFormat()) && originalComponent.equals(event.preview()) && event.getPlayer().getName().equalsIgnoreCase(event.getPlayer().getDisplayName())) { // PaperDragon - compare Adventure Component instead of String message
                     return ichatbasecomponent;
                 }
 
-                return CraftChatMessage.fromStringOrNull(String.format(event.getFormat(), event.getPlayer().getDisplayName(), event.getMessage()));
+                return io.papermc.paper.adventure.PaperAdventure.asVanilla(event.preview()); // PaperDragon - use Adventure Component for preview instead of String
             }, chatExecutor);
         };
         // CraftBukkit end
